// @shopify/hydrogen/dist/production/index.cjs
'use strict';

var react = require('react');
var react$1 = require('@remix-run/react');
var jsxRuntime = require('react/jsx-runtime');
var hydrogenReact = require('@shopify/hydrogen-react');
var cookie = require('worktop/cookie');
var ma = require('content-security-policy-builder');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var ma__default = /*#__PURE__*/_interopDefault(ma);

var Ao=Object.defineProperty;var To=(e,t)=>()=>(e&&(t=e(e=0)),t);var vo=(e,t)=>{for(var r in t)Ao(e,r,{get:t[r],enumerable:!0});};var mo={};vo(mo,{default:()=>yo,logSeoTags:()=>fo});function yo({headTags:e}){return fo(e),null}function fo(e){console.log(" "),console.log("%cSEO Meta Tags",`${ts}`),console.log(" "),e.forEach(t=>{if(t.tag==="script"){if(console.log("%c\u2022 JSON LD ",jt),t.children)try{console.table(JSON.parse(t.children),["name","content"]);}catch{console.log(t.children);}}else {if(console.log(`%c\u2022 ${t.tag} `,jt),t.children)if(typeof t.children=="string")console.log(`\u21B3 ${t.children}`);else try{Object.entries(JSON.parse(t.children)).map(([r,o])=>console.log(`\u21B3 ${o}`));}catch{console.log(t.children);}if(t.props.property==="og:image:url"){let r=t.props.content;rs(r).then(o=>{let n=`font-size: 400px; padding: 10px; background: white url(${o}) no-repeat center; background-size: contain;`;console.log("%c\u2022 Share image preview",jt),console.log("%c  ",n),console.log(`\u21B3 ${r}`);}).catch(o=>{console.error(o);});}Object.entries(t.props).map(([r,o])=>{console.log(`\u21B3 ${r} \u2192 ${o}`);});}console.log(" ");});}async function rs(e){let o=await(await(await fetch(e)).blob()).arrayBuffer();return `data:image/png;base64,${os(o)}`}function os(e){let t="",r=new Uint8Array(e),o=r.byteLength;for(let n=0;n<o;n++)t+=String.fromCharCode(r[n]);return btoa(t)}var jt,ts,go=To(()=>{jt="text-transform: uppercase;",ts="text-transform: uppercase; font-weight: bold; text-transform: uppercase;font-weight: bold";});function ue(e){let{type:t,data:r={},customData:o}=e,n=react$1.useLocation(),{publish:a,cart:s,prevCart:c,shop:i,customData:u}=B(),p=n.pathname+n.search,l={...r,customData:{...u,...o},cart:s,prevCart:c,shop:i};return react.useEffect(()=>{i?.shopId&&(l={...l,url:window.location.href},a(t,l));},[a,p,i?.shopId]),null}function Yt(e){return jsxRuntime.jsx(ue,{...e,type:"page_viewed"})}function Xt(e){return jsxRuntime.jsx(ue,{...e,type:"product_viewed"})}function Zt(e){return jsxRuntime.jsx(ue,{...e,type:"collection_viewed"})}function er(e){return jsxRuntime.jsx(ue,{...e,type:"cart_viewed"})}function tr(e){return jsxRuntime.jsx(ue,{...e,type:"search_viewed"})}function rr(e){return jsxRuntime.jsx(ue,{...e})}var M={PAGE_VIEWED:"page_viewed",PRODUCT_VIEWED:"product_viewed",COLLECTION_VIEWED:"collection_viewed",CART_VIEWED:"cart_viewed",SEARCH_VIEWED:"search_viewed",CART_UPDATED:"cart_updated",PRODUCT_ADD_TO_CART:"product_added_to_cart",PRODUCT_REMOVED_FROM_CART:"product_removed_from_cart",CUSTOM_EVENT:"custom_"};var xo="https://cdn.shopify.com/shopifycloud/consent-tracking-api/v0.1/consent-tracking-api.js",Do="https://cdn.shopify.com/shopifycloud/privacy-banner/storefront-banner.js";function tt(e){console.error(`[h2:error:useCustomerPrivacy] Unable to setup Customer Privacy API: Missing consent.${e} configuration.`);}function rt(e){let{withPrivacyBanner:t=!1,onVisitorConsentCollected:r,onReady:o,...n}=e;hydrogenReact.useLoadScript(t?Do:xo,{attributes:{id:"customer-privacy-api"}});let{observing:a,setLoaded:s}=_o({withPrivacyBanner:t,onLoaded:o}),c=react.useMemo(()=>{let{checkoutDomain:u,storefrontAccessToken:p}=n;return u||tt("checkoutDomain"),p||tt("storefrontAccessToken"),(p.startsWith("shpat_")||p.length!==32)&&console.error("[h2:error:useCustomerPrivacy] It looks like you passed a private access token, make sure to use the public token"),{checkoutRootDomain:u,storefrontAccessToken:p,storefrontRootDomain:ar(u),country:n.country,locale:n.locale}},[n,ar,tt]);react.useEffect(()=>{let u=p=>{r&&r(p.detail);};return document.addEventListener("visitorConsentCollected",u),()=>{document.removeEventListener("visitorConsentCollected",u);}},[r]),react.useEffect(()=>{if(!t||a.current.privacyBanner)return;a.current.privacyBanner=!0;let u=window.privacyBanner||void 0;Object.defineProperty(window,"privacyBanner",{configurable:!0,get(){return u},set(l){if(typeof l=="object"&&l!==null&&"showPreferences"in l&&"loadBanner"in l){let m=l;m.loadBanner(c),u=ir({privacyBanner:m,config:c}),s.privacyBanner(),nr();}}});},[t,c,ir,s.privacyBanner]),react.useEffect(()=>{if(a.current.customerPrivacy)return;a.current.customerPrivacy=!0;let u=null,p=window.Shopify||void 0;Object.defineProperty(window,"Shopify",{configurable:!0,get(){return p},set(l){typeof l=="object"&&l!==null&&Object.keys(l).length===0&&(p=l,Object.defineProperty(window.Shopify,"customerPrivacy",{configurable:!0,get(){return u},set(m){if(typeof m=="object"&&m!==null&&"setTrackingConsent"in m){let y=m;u={...y,setTrackingConsent:sr({customerPrivacy:y,config:c})},p={...p,customerPrivacy:u},s.customerPrivacy(),nr();}}}));}});},[c,sr,s.customerPrivacy]);let i={customerPrivacy:de()};return t&&(i.privacyBanner=Ue()),i}var or=!1;function nr(){if(or)return;or=!0;let e=new CustomEvent("shopifyCustomerPrivacyApiLoaded");document.dispatchEvent(e);}function _o({withPrivacyBanner:e,onLoaded:t}){let r=react.useRef({customerPrivacy:!1,privacyBanner:!1}),[o,n]=react.useState(e?[!1,!1]:[!1]),a=o.every(Boolean),s={customerPrivacy:()=>{n(e?c=>[!0,c[1]]:()=>[!0]);},privacyBanner:()=>{e&&n(c=>[c[0],!0]);}};return react.useEffect(()=>{a&&t&&t();},[a,t]),{observing:r,setLoaded:s}}function ar(e){if(typeof window>"u")return;let t=window.document.location.host,r=e.split(".").reverse(),o=t.split(".").reverse(),n=[];return r.forEach((a,s)=>{a===o[s]&&n.push(a);}),n.reverse().join(".")}function sr({customerPrivacy:e,config:t}){let r=e.setTrackingConsent,{locale:o,country:n,...a}=t;function s(c,i){r({...a,headlessStorefront:!0,...c},i);}return s}function ir({privacyBanner:e,config:t}){let r=e.loadBanner,o=e.showPreferences;function n(s){if(typeof s=="object"){r({...t,...s});return}r(t);}function a(s){if(typeof s=="object"){o({...t,...s});return}o(t);}return {loadBanner:n,showPreferences:a}}function de(){try{return window.Shopify&&window.Shopify.customerPrivacy?window.Shopify?.customerPrivacy:null}catch{return null}}function Ue(){try{return window&&window?.privacyBanner?window.privacyBanner:null}catch{return null}}var cr="2025.1.2";function Mo(){let e=de();if(!e)throw new Error("Shopify Customer Privacy API not available. Must be used within a useEffect. Make sure to load the Shopify Customer Privacy API with useCustomerPrivacy() or <AnalyticsProvider>.");return e}function dr({consent:e,onReady:t,domain:r}){let{subscribe:o,register:n,canTrack:a}=B(),[s,c]=react.useState(!1),[i,u]=react.useState(!1),p=react.useRef(!1),{checkoutDomain:l,storefrontAccessToken:m,language:y}=e,{ready:d}=n("Internal_Shopify_Analytics");return rt({...e,locale:y,checkoutDomain:l||"mock.shop",storefrontAccessToken:m||"abcdefghijklmnopqrstuvwxyz123456",onVisitorConsentCollected:()=>u(!0),onReady:()=>u(!0)}),hydrogenReact.useShopifyCookies({hasUserConsent:i?a():!0,domain:r,checkoutDomain:l}),react.useEffect(()=>{p.current||(p.current=!0,o(M.PAGE_VIEWED,$o),o(M.PRODUCT_VIEWED,qo),o(M.COLLECTION_VIEWED,Fo),o(M.SEARCH_VIEWED,Ho),o(M.PRODUCT_ADD_TO_CART,Bo),c(!0));},[o]),react.useEffect(()=>{s&&i&&(d(),t());},[s,i,t]),null}function ke(e){console.error(`[h2:error:ShopifyAnalytics] Unable to send Shopify analytics: Missing shop.${e} configuration.`);}function Se(e){let t=Mo(),r=t.analyticsProcessingAllowed();if(!e?.shop?.shopId){ke("shopId");return}if(!e?.shop?.acceptedLanguage){ke("acceptedLanguage");return}if(!e?.shop?.currency){ke("currency");return}if(!e?.shop?.hydrogenSubchannelId){ke("hydrogenSubchannelId");return}return {shopifySalesChannel:"hydrogen",assetVersionId:cr,...e.shop,hasUserConsent:r,...hydrogenReact.getClientBrowserParameters(),ccpaEnforced:!t.saleOfDataAllowed(),gdprEnforced:!(t.marketingAllowed()&&t.analyticsProcessingAllowed()),analyticsAllowed:t.analyticsProcessingAllowed(),marketingAllowed:t.marketingAllowed(),saleOfDataAllowed:t.saleOfDataAllowed()}}function Vo(e,t){if(t===null)return;let r=Se(e);return r?{...r,cartId:t.id}:void 0}var Z={};function $o(e){let t=Se(e);t&&(hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.PAGE_VIEW_2,payload:{...t,...Z}}),Z={});}function qo(e){let t=Se(e);if(t&&lr({type:"product",products:e.products})){let r=ot(e.products);Z={pageType:hydrogenReact.AnalyticsPageType.product,resourceId:r[0].productGid},t={...t,...Z,products:ot(e.products)},hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.PRODUCT_VIEW,payload:t});}}function Fo(e){let t=Se(e);t&&(Z={pageType:hydrogenReact.AnalyticsPageType.collection,resourceId:e.collection.id},t={...t,...Z,collectionHandle:e.collection.handle,collectionId:e.collection.id},hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.COLLECTION_VIEW,payload:t}));}function Ho(e){let t=Se(e);t&&(Z={pageType:hydrogenReact.AnalyticsPageType.search},t={...t,...Z,searchString:e.searchTerm},hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.SEARCH_VIEW,payload:t}));}function Bo(e){let{cart:t,currentLine:r}=e,o=Vo(e,t);!o||!r?.id||Go({matchedLine:r,eventPayload:o});}function Go({matchedLine:e,eventPayload:t}){let r={id:e.merchandise.product.id,variantId:e.merchandise.id,title:e.merchandise.product.title,variantTitle:e.merchandise.title,vendor:e.merchandise.product.vendor,price:e.merchandise.price.amount,quantity:e.quantity,productType:e.merchandise.product.productType,sku:e.merchandise.sku};lr({type:"cart",products:[r]})&&hydrogenReact.sendShopifyAnalytics({eventName:hydrogenReact.AnalyticsEventName.ADD_TO_CART,payload:{...t,products:ot([r])}});}function ae(e,t,r,o){if(e==="cart"){let n=`${r?"merchandise":"merchandise.product"}.${t}`;console.error(`[h2:error:ShopifyAnalytics] Can't set up cart analytics events because the \`cart.lines[].${n}\` value is missing from your GraphQL cart query. In your project, search for where \`fragment CartLine on CartLine\` is defined and make sure \`${n}\` is part of your cart query. Check the Hydrogen Skeleton template for reference: https://github.com/Shopify/hydrogen/blob/main/templates/skeleton/app/lib/fragments.ts#L25-L56.`);}else {let n=`${o||t}`;console.error(`[h2:error:ShopifyAnalytics] Can't set up product view analytics events because the \`${n}\` is missing from your \`<Analytics.ProductView>\`. Make sure \`${n}\` is part of your products data prop. Check the Hydrogen Skeleton template for reference: https://github.com/Shopify/hydrogen/blob/main/templates/skeleton/app/routes/products.%24handle.tsx#L159-L165.`);}}function lr({type:e,products:t}){return !t||t.length===0?(ae(e,"",!1,"data.products"),!1):(t.forEach(r=>{if(!r.id)return ae(e,"id",!1),!1;if(!r.title)return ae(e,"title",!1),!1;if(!r.price)return ae(e,"price.amount",!0,"price"),!1;if(!r.vendor)return ae(e,"vendor",!1),!1;if(!r.variantId)return ae(e,"id",!0,"variantId"),!1;if(!r.variantTitle)return ae(e,"title",!0,"variantTitle"),!1}),!0)}function ot(e){return e.map(t=>{let r={productGid:t.id,variantGid:t.variantId,name:t.title,variantName:t.variantTitle,brand:t.vendor,price:t.price,quantity:t.quantity||1,category:t.productType};return t.sku&&(r.sku=t.sku),t.productType&&(r.category=t.productType),r})}function mr(e){console.error(`[h2:error:CartAnalytics] Can't set up cart analytics events because the \`cart.${e}\` value is missing from your GraphQL cart query. In your project, search for where \`fragment CartApiQuery on Cart\` is defined and make sure \`${e}\` is part of your cart query. Check the Hydrogen Skeleton template for reference: https://github.com/Shopify/hydrogen/blob/main/templates/skeleton/app/lib/fragments.ts#L59.`);}function gr({cart:e,setCarts:t}){let{publish:r,shop:o,customData:n,canTrack:a,cart:s,prevCart:c}=B(),i=react.useRef(null);return react.useEffect(()=>{if(e)return Promise.resolve(e).then(u=>{if(u&&u.lines){if(!u.id){mr("id");return}if(!u.updatedAt){mr("updatedAt");return}}t(({cart:p,prevCart:l})=>u?.updatedAt!==p?.updatedAt?{cart:u,prevCart:p}:{cart:p,prevCart:l});}),()=>{}},[t,e]),react.useEffect(()=>{if(!s||!s?.updatedAt||s?.updatedAt===c?.updatedAt)return;let u;try{u=JSON.parse(localStorage.getItem("cartLastUpdatedAt")||"");}catch{u=null;}if(s.id===u?.id&&s.updatedAt===u?.updatedAt)return;let p={eventTimestamp:Date.now(),cart:s,prevCart:c,shop:o,customData:n};if(s.updatedAt===i.current)return;i.current=s.updatedAt,r("cart_updated",p),localStorage.setItem("cartLastUpdatedAt",JSON.stringify({id:s.id,updatedAt:s.updatedAt}));let l=c?.lines?hydrogenReact.flattenConnection(c?.lines):[],m=s.lines?hydrogenReact.flattenConnection(s.lines):[];l?.forEach(y=>{let d=m.filter(g=>y.id===g.id);if(d?.length===1){let g=d[0];y.quantity<g.quantity?r("product_added_to_cart",{...p,prevLine:y,currentLine:g}):y.quantity>g.quantity&&r("product_removed_from_cart",{...p,prevLine:y,currentLine:g});}else r("product_removed_from_cart",{...p,prevLine:y});}),m?.forEach(y=>{let d=l.filter(g=>y.id===g.id);(!d||d.length===0)&&r("product_added_to_cart",{...p,currentLine:y});});},[s,c,r,o,n,a]),null}var Jo="https://cdn.shopify.com/shopifycloud/perf-kit/shopify-perf-kit-1.0.1.min.js";function hr({shop:e}){let t=react.useRef(!1),{subscribe:r,register:o}=B(),{ready:n}=o("Internal_Shopify_Perf_Kit"),a=hydrogenReact.useLoadScript(Jo,{attributes:{id:"perfkit","data-application":"hydrogen","data-shop-id":hydrogenReact.parseGid(e.shopId).id.toString(),"data-storefront-id":e.hydrogenSubchannelId,"data-monorail-region":"global","data-spa-mode":"true","data-resource-timing-sampling-rate":"100"}});return react.useEffect(()=>{a!=="done"||t.current||(t.current=!0,r(M.PAGE_VIEWED,()=>{window.PerfKit?.navigate();}),r(M.PRODUCT_VIEWED,()=>{window.PerfKit?.setPageType("product");}),r(M.COLLECTION_VIEWED,()=>{window.PerfKit?.setPageType("collection");}),r(M.SEARCH_VIEWED,()=>{window.PerfKit?.setPageType("search");}),r(M.CART_VIEWED,()=>{window.PerfKit?.setPageType("cart");}),n());},[r,n,a]),null}var Cr=new Set,G=e=>{Cr.has(e)||(console.warn(e),Cr.add(e));},Pr=new Set,at=e=>{Pr.has(e)||(console.error(new Error(e)),Pr.add(e));};var rn={canTrack:()=>!1,cart:null,customData:{},prevCart:null,publish:()=>{},shop:null,subscribe:()=>{},register:()=>({ready:()=>{}}),customerPrivacy:null,privacyBanner:null},Er=react.createContext(rn),Ve=new Map,Ae={};function br(){return Object.values(Ae).every(Boolean)}function Sr(e,t){Ve.has(e)||Ve.set(e,new Map),Ve.get(e)?.set(t.toString(),t);}var $e=new Map;function Ar(e,t){if(!br()){$e.set(e,t);return}Rr(e,t);}function Rr(e,t){(Ve.get(e)??new Map).forEach((r,o)=>{try{r(t);}catch(n){typeof n=="object"&&n instanceof Error?console.error("Analytics publish error",n.message,o,n.stack):console.error("Analytics publish error",n,o);}});}function Tr(e){return Ae.hasOwnProperty(e)||(Ae[e]=!1),{ready:()=>{Ae[e]=!0,br()&&$e.size>0&&($e.forEach((t,r)=>{Rr(r,t);}),$e.clear());}}}function vr(){try{return window.Shopify.customerPrivacy.analyticsProcessingAllowed()}catch{}return !1}function Ir(e,t){return `[h2:error:Analytics.Provider] - ${e} is required. Make sure ${t} is defined in your environment variables. See https://h2o.fyi/analytics/consent to learn how to setup environment variables in the Shopify admin.`}function on({canTrack:e,cart:t,children:r,consent:o,customData:n={},shop:a=null,cookieDomain:s}){let c=react.useRef(!1),{shop:i}=nn(a),[u,p]=react.useState(!!e),[l,m]=react.useState({cart:null,prevCart:null}),[y,d]=react.useState(e?()=>e:()=>vr);if(i)if(/\/68817551382$/.test(i.shopId))G("[h2:error:Analytics.Provider] - Mock shop is used. Analytics will not work properly.");else {if(!o.checkoutDomain){let P=Ir("consent.checkoutDomain","PUBLIC_CHECKOUT_DOMAIN");at(P);}if(!o.storefrontAccessToken){let P=Ir("consent.storefrontAccessToken","PUBLIC_STOREFRONT_API_TOKEN");at(P);}o?.country||(o.country="US"),o?.language||(o.language="EN"),o.withPrivacyBanner===void 0&&(o.withPrivacyBanner=!1);}let g=react.useMemo(()=>({canTrack:y,...l,customData:n,publish:y()?Ar:()=>{},shop:i,subscribe:Sr,register:Tr,customerPrivacy:de(),privacyBanner:Ue()}),[u,y,l,l.cart?.updatedAt,l.prevCart,Ar,Sr,n,i,Tr,JSON.stringify(Ae),de,Ue]);return jsxRuntime.jsxs(Er.Provider,{value:g,children:[r,!!i&&jsxRuntime.jsx(Yt,{}),!!i&&!!t&&jsxRuntime.jsx(gr,{cart:t,setCarts:m}),!!i&&o.checkoutDomain&&jsxRuntime.jsx(dr,{consent:o,onReady:()=>{c.current=!0,p(!0),d(e?()=>e:()=>vr);},domain:s}),!!i&&jsxRuntime.jsx(hr,{shop:i})]})}function B(){let e=react.useContext(Er);if(!e)throw new Error("[h2:error:useAnalytics] 'useAnalytics()' must be a descendent of <AnalyticsProvider/>");return e}function nn(e){let[t,r]=react.useState(null);return react.useEffect(()=>(Promise.resolve(e).then(r),()=>{}),[r,e]),{shop:t}}async function an({storefront:e,publicStorefrontId:t="0"}){return e.query(sn,{cache:e.CacheLong()}).then(({shop:r,localization:o})=>({shopId:r.id,acceptedLanguage:o.language.isoCode,currency:o.country.currency.isoCode,hydrogenSubchannelId:t}))}var sn=`#graphql
  query ShopData(
    $country: CountryCode
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    shop {
      id
    }
    localization {
      country {
        currency {
          isoCode
        }
      }
      language {
        isoCode
      }
    }
  }
`,cn={CartView:er,CollectionView:Zt,CustomView:rr,ProductView:Xt,Provider:on,SearchView:tr};function Q(e,t){return st(e.headers,t)}function st(e,t){let r=e?.get?.(t)??e?.[t];return typeof r=="string"?r:null}function ee(e){return {requestId:e?Q(e,"request-id"):void 0,purpose:e?Q(e,"purpose"):void 0}}function le(e,t={}){let r=new Error,o=(n,a="Error")=>{let s=(r.stack??"").split(`
`).slice(3+(t.stackOffset??0)).join(`
`).replace(/ at loader(\d+) \(/,(c,i)=>c.replace(i,""));return `${a}: ${n}
`+s};return e.then(n=>{if(n?.errors&&Array.isArray(n.errors)){let a=typeof t.logErrors=="function"?t.logErrors:()=>t.logErrors??!1;n.errors.forEach(s=>{s&&(s.stack=o(s.message,s.name),a(s)&&console.error(s));});}return n}).catch(n=>{throw n&&(n.stack=o(n.message,n.name)),n})}var q=void 0;var qe="public",pn="private",it="no-store",wr={maxAge:"max-age",staleWhileRevalidate:"stale-while-revalidate",sMaxAge:"s-maxage",staleIfError:"stale-if-error"};function Te(e){let t=[];return Object.keys(e).forEach(r=>{r==="mode"?t.push(e[r]):wr[r]&&t.push(`${wr[r]}=${e[r]}`);}),t.join(", ")}function ct(){return {mode:it}}function ut(e){if(e?.mode&&e?.mode!==qe&&e?.mode!==pn)throw Error("'mode' must be either 'public' or 'private'")}function se(e){return ut(e),{mode:qe,maxAge:1,staleWhileRevalidate:9,...e}}function pt(e){return ut(e),{mode:qe,maxAge:3600,staleWhileRevalidate:82800,...e}}function ye(e){return ut(e),{mode:qe,maxAge:1,staleWhileRevalidate:86399,...e}}function dt(e){return e}function W(e){return String(e).includes("__proto__")?JSON.parse(e,dn):JSON.parse(e)}function dn(e,t){if(e!=="__proto__")return t}function Fe(e,t){return e&&t?{...e,...t}:e||ye()}function lt(e){return Te(Fe(e))}async function ln(e,t){if(!e)return;let r=await e.match(t);if(!r){return}return r}async function yn(e,t,r,o){if(!e)return;let n=Fe(o),a=lt(Fe(n,{maxAge:(n.maxAge||0)+(n.staleWhileRevalidate||0)})),s=lt(Fe(n));r.headers.set("cache-control",a),r.headers.set("real-cache-control",s),r.headers.set("cache-put-date",String(Date.now())),await e.put(t,r);}async function fn(e,t){e&&await e.delete(t);}function mn(e,t){let r=e.headers.get("real-cache-control"),o=0;if(r){let a=r.match(/max-age=(\d*)/);a&&a.length>1&&(o=parseFloat(a[1]));}return [(Date.now()-Number(t))/1e3,o]}function gn(e,t){let r=t.headers.get("cache-put-date");if(!r)return !1;let[o,n]=mn(t,r),a=o>n;return a}var He={get:ln,set:yn,delete:fn,generateDefaultCacheControlHeader:lt,isStale:gn};function ve(e){return `https://shopify.dev/?${e}`}function hn(e){return e||ye()}async function Or(e,t){if(!e)return;let r=ve(t),o=new Request(r),n=await He.get(e,o);if(!n)return;let a=await n.text();try{return [W(a),n]}catch{return [a,n]}}async function xr(e,t,r,o){if(!e)return;let n=ve(t),a=new Request(n),s=new Response(JSON.stringify(r));await He.set(e,a,s,hn(o));}function Dr(e,t){return He.isStale(new Request(ve(e)),t)}function _r(e){let t=Array.isArray(e)?e:[e],r="";for(let o of t)o!=null&&(typeof o=="object"?r+=JSON.stringify(o):r+=o.toString());return encodeURIComponent(r)}var yt=new Set;async function Be(e,t,{strategy:r=se(),cacheInstance:o,shouldCacheResult:n=()=>!0,waitUntil:a,debugInfo:s}){let i=_r([...typeof e=="string"?[e]:e]),l=f=>{({displayName:f.displayName,url:f.response?.url,responseInit:{status:f.response?.status||0,statusText:f.response?.statusText||"",headers:Array.from(f.response?.headers.entries()||[])}});},y=void 0;if(!o||!r||r.mode===it){let f=await t({addDebugData:l});return f}let d=f=>xr(o,i,{value:f,debugInfo:void 0},r),g=await Or(o,i);if(g&&typeof g[0]!="string"){let[{value:f,debugInfo:h},A]=g;let R=Dr(i,A)?"STALE":"HIT";if(!yt.has(i)&&R==="STALE"){yt.add(i);let D=Promise.resolve().then(async()=>{let U=Date.now();try{let O=await t({addDebugData:l});n(O)&&(await d(O),y?.({result:O,cacheStatus:"PUT",overrideStartTime:U}));}catch(O){O.message&&(O.message="SWR in sub-request failed: "+O.message),console.error(O);}finally{yt.delete(i);}});a?.(D);}return f}let P=await t({addDebugData:l});if(n(P)){let f=Promise.resolve().then(async()=>{await d(P);});a?.(f);}return P}function Cn(e,t){return [e,{status:t.status,statusText:t.statusText,headers:Array.from(t.headers.entries())}]}function Lr([e,t]){return [e,new Response(e,t)]}async function Ge(e,t,{cacheInstance:r,cache:o,cacheKey:n=[e,t],shouldCacheResponse:a,waitUntil:s,debugInfo:c}){return !o&&(!t.method||t.method==="GET")&&(o=se()),Be(n,async()=>{let i=await fetch(e,t);if(!i.ok)return i;let u=await i.text().catch(()=>"");try{u&&(u=W(u));}catch{}return Cn(u,i)},{cacheInstance:r,waitUntil:s,strategy:o??null,debugInfo:c,shouldCacheResult:i=>"ok"in i?!1:a(...Lr(i))}).then(i=>"ok"in i?[null,i]:Lr(i))}function Pn(e){let{cache:t,waitUntil:r,request:o}=e;return {run:({cacheKey:n,cacheStrategy:a,shouldCacheResult:s},c)=>Be(n,c,{shouldCacheResult:s,strategy:a,cacheInstance:t,waitUntil:r,debugInfo:{...ee(o),stackInfo:q?.()}}),fetch:(n,a,s)=>Ge(n,a??{},{waitUntil:r,cacheKey:[n,a],cacheInstance:t,debugInfo:{url:n,...ee(o),stackInfo:q?.(),displayName:s?.displayName},cache:s.cacheStrategy,...s}).then(([c,i])=>({data:c,response:i}))}}var ft=class{#e;constructor(){this.#e=new Map;}add(t){throw new Error("Method not implemented. Use `put` instead.")}addAll(t){throw new Error("Method not implemented. Use `put` instead.")}matchAll(t,r){throw new Error("Method not implemented. Use `match` instead.")}async put(t,r){if(t.method!=="GET")throw new TypeError("Cannot cache response to non-GET request.");if(r.status===206)throw new TypeError("Cannot cache response to a range request (206 Partial Content).");if(r.headers.get("vary")?.includes("*"))throw new TypeError("Cannot cache response with 'Vary: *' header.");this.#e.set(t.url,{body:new Uint8Array(await r.arrayBuffer()),status:r.status,headers:[...r.headers],timestamp:Date.now()});}async match(t){if(t.method!=="GET")return;let r=this.#e.get(t.url);if(!r)return;let{body:o,timestamp:n,...a}=r,s=new Headers(a.headers),c=s.get("cache-control")||s.get("real-cache-control")||"",i=parseInt(c.match(/max-age=(\d+)/)?.[1]||"0",10),u=parseInt(c.match(/stale-while-revalidate=(\d+)/)?.[1]||"0",10),p=(Date.now()-n)/1e3;if(p>i+u){this.#e.delete(t.url);return}let m=p>i;return s.set("cache",m?"STALE":"HIT"),s.set("date",new Date(n).toUTCString()),new Response(o,{status:a.status??200,headers:s})}async delete(t){return this.#e.has(t.url)?(this.#e.delete(t.url),!0):!1}keys(t){let r=[];for(let o of this.#e.keys())(!t||t.url===o)&&r.push(new Request(o));return Promise.resolve(r)}};var Ur="cartFormInput";function j({children:e,action:t,inputs:r,route:o,fetcherKey:n}){let a=react$1.useFetcher({key:n});return jsxRuntime.jsxs(a.Form,{action:o||"",method:"post",children:[(t||r)&&jsxRuntime.jsx("input",{type:"hidden",name:Ur,value:JSON.stringify({action:t,inputs:r})}),typeof e=="function"?e(a):e]})}j.INPUT_NAME=Ur;j.ACTIONS={AttributesUpdateInput:"AttributesUpdateInput",BuyerIdentityUpdate:"BuyerIdentityUpdate",Create:"Create",DiscountCodesUpdate:"DiscountCodesUpdate",GiftCardCodesUpdate:"GiftCardCodesUpdate",LinesAdd:"LinesAdd",LinesRemove:"LinesRemove",LinesUpdate:"LinesUpdate",NoteUpdate:"NoteUpdate",SelectedDeliveryOptionsUpdate:"SelectedDeliveryOptionsUpdate",MetafieldsSet:"MetafieldsSet",MetafieldDelete:"MetafieldDelete"};function An(e){let t={};for(let s of e.entries()){let c=s[0],i=e.getAll(c);t[c]=i.length>1?i:s[1];}let{cartFormInput:r,...o}=t,{action:n,inputs:a}=r?JSON.parse(String(r)):{};return {action:n,inputs:{...a,...o}}}j.getFormInput=An;var mt=e=>{let t=cookie.parse(st(e,"Cookie")||"");return ()=>t.cart?`gid://shopify/Cart/${t.cart}`:void 0};var gt=e=>t=>{let r=new Headers;return r.append("Set-Cookie",cookie.stringify("cart",t.split("/").pop()||"",{path:"/",...e})),r};var ht="Custom-Storefront-Request-Group-ID",Ct="X-Shopify-Storefront-Access-Token",Pt="X-SDK-Variant",St="X-SDK-Variant-Source",At="X-SDK-Version";function kr(){return typeof crypto<"u"&&crypto.randomUUID?crypto.randomUUID():`weak-${Math.random().toString(16).substring(2)}`}var Ie="2025.1.2";function fe(e){return e.replace(/\s*#.*$/gm,"").replace(/\s+/gm," ").trim()}var bn=/(^|}\s)query[\s({]/im,Rn=/(^|}\s)mutation[\s({]/im;function Qe(e,t){if(!bn.test(e))throw new Error(`[h2:error:${t}] Can only execute queries`)}function We(e,t){if(!Rn.test(e))throw new Error(`[h2:error:${t}] Can only execute mutations`)}var ie=class extends Error{locations;path;extensions;constructor(t,r={}){let n=(r.clientOperation?`[h2:error:${r.clientOperation}] `:"")+t+(r.requestId?` - Request ID: ${r.requestId}`:"");super(n),this.name="GraphQLError",this.extensions=r.extensions,this.locations=r.locations,this.path=r.path,this.stack=r.stack||void 0;try{this.cause=JSON.stringify({...typeof r.cause=="object"?r.cause:{},requestId:r.requestId});}catch{r.cause&&(this.cause=r.cause);}}get[Symbol.toStringTag](){return this.name}toString(){let t=`${this.name}: ${this.message}`;if(this.path)try{t+=` | path: ${JSON.stringify(this.path)}`;}catch{}if(this.extensions)try{t+=` | extensions: ${JSON.stringify(this.extensions)}`;}catch{}return t+=`
`,this.stack&&(t+=`${this.stack.slice(this.stack.indexOf(`
`)+1)}
`),t}toJSON(){return {name:"Error",message:""}}};function Ee({url:e,response:t,errors:r,type:o,query:n,queryVariables:a,ErrorConstructor:s=Error,client:c="storefront"}){let i=(typeof r=="string"?r:r?.map?.(p=>p.message).join(`
`))||`URL: ${e}
API response error: ${t.status}`,u=new ie(i,{query:n,queryVariables:a,cause:{errors:r},clientOperation:`${c}.${o}`,requestId:t.headers.get("x-request-id")});throw new s(u.message,{cause:u.cause})}var Ln={language:"EN",country:"US"};function $r(e){let {storefrontHeaders:t,cache:r,waitUntil:o,i18n:n,storefrontId:a,logErrors:s=!0,...c}=e,{getPublicTokenHeaders:u,getPrivateTokenHeaders:p,getStorefrontApiUrl:l,getShopifyDomain:m}=hydrogenReact.createStorefrontClient(c),d=(c.privateStorefrontToken?p:u)({contentType:"json",buyerIp:t?.buyerIp||""});if(d[ht]=t?.requestGroupId||kr(),a&&(d[hydrogenReact.SHOPIFY_STOREFRONT_ID_HEADER]=a),(d["user-agent"]=`Hydrogen ${Ie}`),t&&t.cookie){let f=hydrogenReact.getShopifyCookies(t.cookie??"");f[hydrogenReact.SHOPIFY_Y]&&(d[hydrogenReact.SHOPIFY_STOREFRONT_Y_HEADER]=f[hydrogenReact.SHOPIFY_Y]),f[hydrogenReact.SHOPIFY_S]&&(d[hydrogenReact.SHOPIFY_STOREFRONT_S_HEADER]=f[hydrogenReact.SHOPIFY_S]);}let g=JSON.stringify({"content-type":d["content-type"],"user-agent":d["user-agent"],[Pt]:d[Pt],[St]:d[St],[At]:d[At],[Ct]:d[Ct]});async function P({query:f,mutation:h,variables:A,cache:R,headers:D=[],storefrontApiVersion:U,displayName:O,stackInfo:F}){let xe=D instanceof Headers?Object.fromEntries(D.entries()):Array.isArray(D)?Object.fromEntries(D):D,oe=f??h,J={...A};n&&(!A?.country&&/\$country/.test(oe)&&(J.country=n.country),!A?.language&&/\$language/.test(oe)&&(J.language=n.language));let ne=l({storefrontApiVersion:U}),C=JSON.stringify({query:oe,variables:J}),S={method:"POST",headers:{...d,...xe},body:C},x=[ne,S.method,g,S.body],[w,V]=await Ge(ne,S,{cacheInstance:h?void 0:r,cache:R||ye(),cacheKey:x,waitUntil:o,shouldCacheResponse:v=>!v?.errors,debugInfo:{requestId:S.headers[ht],displayName:O,url:ne,stackInfo:F,graphql:C,purpose:t?.purpose}}),H={url:ne,response:V,type:h?"mutation":"query",query:oe,queryVariables:J,errors:void 0};if(!V.ok){let v,N=w;try{N??=await V.text(),v=W(N);}catch{v=[{message:N??"Could not parse Storefront API response"}];}Ee({...H,errors:v});}let{data:k,errors:Y}=w,X=Y?.map(({message:v,...N})=>new ie(v,{...N,clientOperation:`storefront.${H.type}`,requestId:V.headers.get("x-request-id"),queryVariables:J,query:oe}));return T(k,X)}return {storefront:{query(f,h){f=fe(f),Qe(f,"storefront.query");let A=Vr?.(f);return le(P({...h,query:f,stackInfo:q?.(A)}),{stackOffset:A,logErrors:s})},mutate(f,h){f=fe(f),We(f,"storefront.mutate");let A=Vr?.(f);return le(P({...h,mutation:f,stackInfo:q?.(A)}),{stackOffset:A,logErrors:s})},cache:r,CacheNone:ct,CacheLong:pt,CacheShort:se,CacheCustom:dt,generateCacheControlHeader:Te,getPublicTokenHeaders:u,getPrivateTokenHeaders:p,getShopifyDomain:m,getApiUrl:l,i18n:n??Ln}}}var Vr=void 0;function T(e,t){return {...e,...t&&{errors:t}}}function Tt({storefront:e,customerAccount:t,getCartId:r,cartFragment:o}){return async n=>{let a=r();if(!a)return null;let[s,{cart:c,errors:i}]=await Promise.all([t?t.isLoggedIn():!1,e.query(Un(o),{variables:{cartId:a,...n},cache:e.CacheNone()})]);if(s&&c?.checkoutUrl){let u=new URL(c.checkoutUrl);u.searchParams.set("logged_in","true"),c.checkoutUrl=u.toString();}return c||i?T(c,i):null}}var Un=(e=kn)=>`#graphql
  query CartQuery(
    $cartId: ID!
    $numCartLines: Int = 100
    $country: CountryCode = ZZ
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    cart(id: $cartId) {
      ...CartApiQuery
    }
  }

  ${e}
`,kn=`#graphql
  fragment CartApiQuery on Cart {
    updatedAt
    id
    checkoutUrl
    totalQuantity
    buyerIdentity {
      countryCode
      customer {
        id
        email
        firstName
        lastName
        displayName
      }
      email
      phone
    }
    lines(first: $numCartLines) {
      edges {
        node {
          id
          quantity
          attributes {
            key
            value
          }
          cost {
            totalAmount {
              amount
              currencyCode
            }
            amountPerQuantity {
              amount
              currencyCode
            }
            compareAtAmountPerQuantity {
              amount
              currencyCode
            }
          }
          merchandise {
            ... on ProductVariant {
              id
              availableForSale
              compareAtPrice {
                ...CartApiMoney
              }
              price {
                ...CartApiMoney
              }
              requiresShipping
              title
              image {
                ...CartApiImage
              }
              product {
                handle
                title
                id
                vendor
              }
              selectedOptions {
                name
                value
              }
            }
          }
        }
      }
    }
    cost {
      subtotalAmount {
        ...CartApiMoney
      }
      totalAmount {
        ...CartApiMoney
      }
      totalDutyAmount {
        ...CartApiMoney
      }
      totalTaxAmount {
        ...CartApiMoney
      }
    }
    note
    attributes {
      key
      value
    }
    discountCodes {
      applicable
      code
    }
  }

  fragment CartApiMoney on MoneyV2 {
    currencyCode
    amount
  }

  fragment CartApiImage on Image {
    id
    url
    altText
    width
    height
  }
`;var I=`#graphql
  fragment CartApiError on CartUserError {
    message
    field
    code
  }
`,E=`#graphql
  fragment CartApiMutation on Cart {
    id
    totalQuantity
    checkoutUrl
  }
`,b=`#graphql
  fragment CartApiWarning on CartWarning {
    code
    message
    target
  }
`;function vt(e){return async(t,r)=>{let o=e.customerAccount?await e.customerAccount.getBuyer():void 0,{cartId:n,...a}=r||{},{buyerIdentity:s,...c}=t,{cartCreate:i,errors:u}=await e.storefront.mutate(Nn(e.cartFragment),{variables:{input:{...c,buyerIdentity:{...o,...s}},...a}});return T(i,u)}}var Nn=(e=E)=>`#graphql
  mutation cartCreate(
    $input: CartInput!
    $country: CountryCode = ZZ
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    cartCreate(input: $input) {
      cart {
        ...CartApiMutation
        checkoutUrl
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${I}
  ${b}
`;function It(e){return async(t,r)=>{let{cartLinesAdd:o,errors:n}=await e.storefront.mutate(Mn(e.cartFragment),{variables:{cartId:e.getCartId(),lines:t,...r}});return T(o,n)}}var Mn=(e=E)=>`#graphql
  mutation cartLinesAdd(
    $cartId: ID!
    $lines: [CartLineInput!]!
    $country: CountryCode = ZZ
    $language: LanguageCode
  ) @inContext(country: $country, language: $language) {
    cartLinesAdd(cartId: $cartId, lines: $lines) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${I}
  ${b}
`;var qr="__h_pending_";function Fr(e){return qr+e}function je(e){return e.startsWith(qr)}function Ke(e,t){if(t.some(r=>je(typeof r=="string"?r:r.id)))throw new Error(`Tried to perform an action on an optimistic line. Make sure to disable your "${e}" CartForm action when the line is optimistic.`)}function Et(e){return async(t,r)=>{Ke("updateLines",t);let{cartLinesUpdate:o,errors:n}=await e.storefront.mutate(Vn(e.cartFragment),{variables:{cartId:e.getCartId(),lines:t,...r}});return T(o,n)}}var Vn=(e=E)=>`#graphql
  mutation cartLinesUpdate(
    $cartId: ID!
    $lines: [CartLineUpdateInput!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartLinesUpdate(cartId: $cartId, lines: $lines) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${I}
  ${b}
`;function bt(e){return async(t,r)=>{Ke("removeLines",t);let{cartLinesRemove:o,errors:n}=await e.storefront.mutate($n(e.cartFragment),{variables:{cartId:e.getCartId(),lineIds:t,...r}});return T(o,n)}}var $n=(e=E)=>`#graphql
  mutation cartLinesRemove(
    $cartId: ID!
    $lineIds: [ID!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartLinesRemove(cartId: $cartId, lineIds: $lineIds) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${I}
  ${b}
`;function Rt(e){return async(t,r)=>{let o=t.filter((s,c,i)=>i.indexOf(s)===c),{cartDiscountCodesUpdate:n,errors:a}=await e.storefront.mutate(qn(e.cartFragment),{variables:{cartId:e.getCartId(),discountCodes:o,...r}});return T(n,a)}}var qn=(e=E)=>`#graphql
  mutation cartDiscountCodesUpdate(
    $cartId: ID!
    $discountCodes: [String!]
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartDiscountCodesUpdate(cartId: $cartId, discountCodes: $discountCodes) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${I}
  ${b}
`;function wt(e){return async(t,r)=>{t.companyLocationId&&e.customerAccount&&e.customerAccount.setBuyer({companyLocationId:t.companyLocationId});let o=e.customerAccount?await e.customerAccount.getBuyer():void 0,{cartBuyerIdentityUpdate:n,errors:a}=await e.storefront.mutate(Fn(e.cartFragment),{variables:{cartId:e.getCartId(),buyerIdentity:{...o,...t},...r}});return T(n,a)}}var Fn=(e=E)=>`#graphql
  mutation cartBuyerIdentityUpdate(
    $cartId: ID!
    $buyerIdentity: CartBuyerIdentityInput!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartBuyerIdentityUpdate(cartId: $cartId, buyerIdentity: $buyerIdentity) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${I}
  ${b}
`;function Ot(e){return async(t,r)=>{let{cartNoteUpdate:o,errors:n}=await e.storefront.mutate(Hn(e.cartFragment),{variables:{cartId:e.getCartId(),note:t,...r}});return T(o,n)}}var Hn=(e=E)=>`#graphql
  mutation cartNoteUpdate(
    $cartId: ID!
    $note: String!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartNoteUpdate(cartId: $cartId, note: $note) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${I}
  ${b}
`;function xt(e){return async(t,r)=>{let{cartSelectedDeliveryOptionsUpdate:o,errors:n}=await e.storefront.mutate(Bn(e.cartFragment),{variables:{cartId:e.getCartId(),selectedDeliveryOptions:t,...r}});return T(o,n)}}var Bn=(e=E)=>`#graphql
  mutation cartSelectedDeliveryOptionsUpdate(
    $cartId: ID!
    $selectedDeliveryOptions: [CartSelectedDeliveryOptionInput!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartSelectedDeliveryOptionsUpdate(cartId: $cartId, selectedDeliveryOptions: $selectedDeliveryOptions) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${I}
  ${b}
`;function Dt(e){return async(t,r)=>{let{cartAttributesUpdate:o,errors:n}=await e.storefront.mutate(Gn(e.cartFragment),{variables:{cartId:r?.cartId||e.getCartId(),attributes:t}});return T(o,n)}}var Gn=(e=E)=>`#graphql
  mutation cartAttributesUpdate(
    $cartId: ID!
    $attributes: [AttributeInput!]!
  ) {
    cartAttributesUpdate(cartId: $cartId, attributes: $attributes) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${I}
  ${b}
`;function _t(e){return async(t,r)=>{let o=r?.cartId||e.getCartId(),n=t.map(c=>({...c,ownerId:o})),{cartMetafieldsSet:a,errors:s}=await e.storefront.mutate(Qn(),{variables:{metafields:n}});return T({cart:{id:o},...a},s)}}var Qn=()=>`#graphql
  mutation cartMetafieldsSet(
    $metafields: [CartMetafieldsSetInput!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartMetafieldsSet(metafields: $metafields) {
      userErrors {
        code
        elementIndex
        field
        message
      }
    }
  }
`;function Lt(e){return async(t,r)=>{let o=r?.cartId||e.getCartId(),{cartMetafieldDelete:n,errors:a}=await e.storefront.mutate(Wn(),{variables:{input:{ownerId:o,key:t}}});return T({cart:{id:o},...n},a)}}var Wn=()=>`#graphql
  mutation cartMetafieldDelete(
    $input: CartMetafieldDeleteInput!
  ) {
    cartMetafieldDelete(input: $input) {
      userErrors {
        code
        field
        message
      }
    }
  }
`;function Ut(e){return async(t,r)=>{let o=t.filter((s,c,i)=>i.indexOf(s)===c),{cartGiftCardCodesUpdate:n,errors:a}=await e.storefront.mutate(jn(e.cartFragment),{variables:{cartId:e.getCartId(),giftCardCodes:o,...r}});return T(n,a)}}var jn=(e=E)=>`#graphql
  mutation cartGiftCardCodesUpdate(
    $cartId: ID!
    $giftCardCodes: [String!]!
    $language: LanguageCode
    $country: CountryCode
  ) @inContext(country: $country, language: $language) {
    cartGiftCardCodesUpdate(cartId: $cartId, giftCardCodes: $giftCardCodes) {
      cart {
        ...CartApiMutation
      }
      userErrors {
        ...CartApiError
      }
      warnings {
        ...CartApiWarning
      }
    }
  }
  ${e}
  ${I}
  ${b}
`;function kt(e){let{getCartId:t,setCartId:r,storefront:o,customerAccount:n,cartQueryFragment:a,cartMutateFragment:s}=e,c=t(),i=()=>c||t(),u={storefront:o,getCartId:i,cartFragment:s,customerAccount:n},p=vt(u),l=async function(...y){let d=await p(...y);return c=d?.cart?.id,d},m={get:Tt({storefront:o,customerAccount:n,getCartId:i,cartFragment:a}),getCartId:i,setCartId:r,create:l,addLines:async(y,d)=>{let g=y.map(P=>({attributes:P.attributes,quantity:P.quantity,merchandiseId:P.merchandiseId,sellingPlanId:P.sellingPlanId}));return c||d?.cartId?await It(u)(g,d):await l({lines:g},d)},updateLines:Et(u),removeLines:bt(u),updateDiscountCodes:async(y,d)=>c||d?.cartId?await Rt(u)(y,d):await l({discountCodes:y},d),updateGiftCardCodes:async(y,d)=>c||d?.cartId?await Ut(u)(y,d):await l({giftCardCodes:y},d),updateBuyerIdentity:async(y,d)=>c||d?.cartId?await wt(u)(y,d):await l({buyerIdentity:y},d),updateNote:async(y,d)=>c||d?.cartId?await Ot(u)(y,d):await l({note:y},d),updateSelectedDeliveryOption:xt(u),updateAttributes:async(y,d)=>c||d?.cartId?await Dt(u)(y,d):await l({attributes:y},d),setMetafields:async(y,d)=>c||d?.cartId?await _t(u)(y,d):await l({metafields:y},d),deleteMetafield:Lt(u)};return "customMethods"in e?{...m,...e.customMethods??{}}:m}function zn(e){let t=react$1.useFetchers();if(!t||!t.length)return e;let r=e?.lines?structuredClone(e):{lines:{nodes:[]}},o=r.lines.nodes,n=!1;for(let{formData:a}of t){if(!a)continue;let s=j.getFormInput(a);if(s.action===j.ACTIONS.LinesAdd)for(let c of s.inputs.lines){if(!c.selectedVariant){console.error("[h2:error:useOptimisticCart] No selected variant was passed in the cart action. Make sure to pass the selected variant if you want to use an optimistic cart");continue}let i=o.find(u=>u.merchandise.id===c.selectedVariant?.id);n=!0,i?(i.quantity=(i.quantity||1)+(c.quantity||1),i.isOptimistic=!0):o.unshift({id:Fr(c.selectedVariant.id),merchandise:c.selectedVariant,isOptimistic:!0,quantity:c.quantity||1});}else if(s.action===j.ACTIONS.LinesRemove)for(let c of s.inputs.lineIds){let i=o.findIndex(u=>u.id===c);if(i!==-1){if(je(o[i].id)){console.error("[h2:error:useOptimisticCart] Tried to remove an optimistic line that has not been added to the cart yet");continue}o.splice(i,1),n=!0;}else console.warn(`[h2:warn:useOptimisticCart] Tried to remove line '${c}' but it doesn't exist in the cart`);}else if(s.action===j.ACTIONS.LinesUpdate)for(let c of s.inputs.lines){let i=o.findIndex(u=>c.id===u.id);if(i>-1){if(je(o[i].id)){console.error("[h2:error:useOptimisticCart] Tried to update an optimistic line that has not been added to the cart yet");continue}o[i].quantity=c.quantity,o[i].quantity===0&&o.splice(i,1),n=!0;}else console.warn(`[h2:warn:useOptimisticCart] Tried to update line '${c.id}' but it doesn't exist in the cart`);}}return n&&(r.isOptimistic=n),r.totalQuantity=o.reduce((a,s)=>a+s.quantity,0),r}var Jn="https://raw.githubusercontent.com/Shopify/hydrogen/main/docs/changelog.json";async function Yn({request:e,changelogUrl:t}){new URL(e.url).searchParams;return fetch(t||Jn)}var ze="2025-01",me=`Shopify Hydrogen ${Ie}`,Hr="30243aa5-17c1-465a-8493-944bcc4e88aa",_="customerAccount",ge="buyer";var $=class extends Response{constructor(t,r,o){super(`Bad request: ${t}`,{status:400,headers:o});}};function Re(e,t={}){let r=t.headers?new Headers(t.headers):new Headers({});return r.set("location",e),new Response(null,{status:t.status||302,headers:r})}async function Xn({session:e,customerAccountId:t,customerAccountTokenExchangeUrl:r,httpsOrigin:o,debugInfo:n}){let a=new URLSearchParams,s=e.get(_),c=s?.refreshToken,i=s?.idToken;if(!c)throw new $("Unauthorized","No refreshToken found in the session. Make sure your session is configured correctly and passed to `createCustomerAccountClient`.");a.append("grant_type","refresh_token"),a.append("refresh_token",c),a.append("client_id",t);let u={"content-type":"application/x-www-form-urlencoded","User-Agent":me,Origin:o};new Date().getTime();let l=r,m=await fetch(l,{method:"POST",headers:u,body:a});if(!m.ok){let f=await m.text();throw new Response(f,{status:m.status,headers:{"Content-Type":"text/html; charset=utf-8"}})}let{access_token:y,expires_in:d,refresh_token:g}=await m.json(),P=await Nt(y,t,r,o);e.set(_,{accessToken:P,expiresAt:new Date(new Date().getTime()+(d-120)*1e3).getTime()+"",refreshToken:g,idToken:i});}function he(e){e.unset(_),e.unset(ge);}async function Br({locks:e,expiresAt:t,session:r,customerAccountId:o,customerAccountTokenExchangeUrl:n,httpsOrigin:a,debugInfo:s}){if(parseInt(t,10)-1e3<new Date().getTime())try{e.refresh||(e.refresh=Xn({session:r,customerAccountId:o,customerAccountTokenExchangeUrl:n,httpsOrigin:a,debugInfo:s})),await e.refresh,delete e.refresh;}catch(c){throw he(r),c&&c.status!==401?c:new $("Unauthorized","Login before querying the Customer Account API.")}}function Gr(){let e=Zn();return Wr(e)}async function Qr(e){let t=await crypto.subtle.digest({name:"SHA-256"},new TextEncoder().encode(e)),r=ea(t);return Wr(r)}function Zn(){let e=new Uint8Array(32);return crypto.getRandomValues(e),String.fromCharCode.apply(null,Array.from(e))}function Wr(e){return btoa(e).replace(/\+/g,"-").replace(/\//g,"_").replace(/=/g,"")}function ea(e){let t=new Uint8Array(e),r=Array.from(t);return String.fromCharCode(...r)}function jr(){let e=Date.now().toString(),t=Math.random().toString(36).substring(2);return e+t}async function Nt(e,t,r,o,n){let a=t;if(!e)throw new $("Unauthorized","oAuth access token was not provided during token exchange.");let s=new URLSearchParams;s.append("grant_type","urn:ietf:params:oauth:grant-type:token-exchange"),s.append("client_id",a),s.append("audience",Hr),s.append("subject_token",e),s.append("subject_token_type","urn:ietf:params:oauth:token-type:access_token"),s.append("scopes","https://api.customers.com/auth/customer.graphql");let c={"content-type":"application/x-www-form-urlencoded","User-Agent":me,Origin:o};new Date().getTime();let u=r,p=await fetch(u,{method:"POST",headers:c,body:s});let l=await p.json();if(l.error)throw new $(l.error_description);return l.access_token}function Kr(e){return ta(e).payload.nonce}function ta(e){let[t,r,o]=e.split("."),n=JSON.parse(atob(t)),a=JSON.parse(atob(r));return {header:n,payload:a,signature:o}}function Je(){return oa(ra())}function ra(){try{return crypto.getRandomValues(new Uint8Array(16))}catch{return new Uint8Array(16).map(()=>Math.random()*255|0)}}function oa(e){return Array.from(e,function(t){return ("0"+(t&255).toString(16)).slice(-2)}).join("")}function Ye(e){if(!e)return;let{pathname:t,search:r}=new URL(e),o=t+r,n=new URLSearchParams(r),a=n.get("return_to")||n.get("redirect");if(a){if(Jr(e,a))return a;console.warn(`Cross-domain redirects are not supported. Tried to redirect from ${o} to ${a}`);}}function Jr(e,t){try{return new URL(e).origin===new URL(t,e).origin}catch{return !1}}function Mt({requestUrl:e,defaultUrl:t,redirectUrl:r}){let o=e,n=zr(e,t),a=r?zr(e,r):n;return Jr(e,a.toString())?a.toString():(console.warn(`Cross-domain redirects are not supported. Tried to redirect from ${o} to ${a}. Default url ${n} is used instead.`),n.toString())}function zr(e,t){return na(t)?new URL(t):new URL(t,new URL(e).origin)}function na(e){try{return new URL(e),!0}catch{return !1}}function Yr(e,t){let r=`https://shopify.com/${t}`,o=`https://shopify.com/authentication/${t}`;return function(a){switch(a){case"CA_BASE_URL":return r;case"CA_BASE_AUTH_URL":return o;case"GRAPHQL":return `${r}/account/customer/api/${e}/graphql`;case"AUTH":return `${o}/oauth/authorize`;case"LOGIN_SCOPE":return t?"openid email customer-account-api:full":"openid email https://api.customers.com/auth/customer.graphql";case"TOKEN_EXCHANGE":return `${o}/oauth/token`;case"LOGOUT":return `${o}/logout`}}}function aa(e,t){if(!e.url)return t;let{pathname:r}=new URL(e.url),o=r.replace(/\.data$/,"").replace(/\/_root$/,"/").replace(/(.+)\/$/,"$1"),n=t+`?${new URLSearchParams({return_to:o}).toString()}`;return Re(n)}function Vt({session:e,customerAccountId:t,shopId:r,customerApiVersion:o=ze,request:n,waitUntil:a,authUrl:s,customAuthStatusHandler:c,logErrors:i=!0,loginPath:u="/account/login",authorizePath:p="/account/authorize",defaultRedirectPath:l="/account",language:m}){if(o!==ze&&console.warn(`[h2:warn:createCustomerAccountClient] You are using Customer Account API version ${o} when this version of Hydrogen was built for ${ze}.`),e||console.warn("[h2:warn:createCustomerAccountClient] session is required to use Customer Account API. Ensure the session object passed in exist."),!n?.url)throw new Error("[h2:error:createCustomerAccountClient] The request object does not contain a URL.");let y=c||(()=>aa(n,u)),d=new URL(n.url),g=d.protocol==="http:"?d.origin.replace("http","https"):d.origin,P=Mt({requestUrl:g,defaultUrl:p,redirectUrl:s}),f=Yr(o,r),h=sa(f,t),A=f("GRAPHQL"),R={};async function D({query:C,type:S,variables:x={}}){let w=await F();if(!w)throw y();new Date().getTime();let k=await fetch(A,{method:"POST",headers:{"Content-Type":"application/json","User-Agent":me,Origin:g,Authorization:w},body:JSON.stringify({query:C,variables:x})});let Y=await k.text(),X={url:A,response:k,type:S,query:C,queryVariables:x,errors:void 0,client:"customer"};if(!k.ok){if(k.status===401)throw he(e),y();let v;try{v=W(Y);}catch{v=[{message:Y}];}Ee({...X,errors:v});}try{let v=W(Y),{errors:N}=v,De=N?.map(({message:_e,...et})=>new ie(_e,{...et,clientOperation:`customerAccount.${X.type}`,requestId:k.headers.get("x-request-id"),queryVariables:x,query:C}));return {...v,...N&&{errors:De}}}catch{Ee({...X,errors:[{message:Y}]});}}async function U(){if(!r)return !1;let C=e.get(_),S=C?.accessToken,x=C?.expiresAt;if(!S||!x)return !1;let w=q?.();try{await Br({locks:R,expiresAt:x,session:e,customerAccountId:t,customerAccountTokenExchangeUrl:f("TOKEN_EXCHANGE"),httpsOrigin:g,debugInfo:{waitUntil:a,stackInfo:w,...ee(n)}});}catch{return !1}return !0}async function O(){if(!await U())throw y()}async function F(){if(await U())return e.get(_)?.accessToken}async function xe(C,S){return h(),C=fe(C),We(C,"customer.mutate"),le(D({query:C,type:"mutation",...S}),{logErrors:i})}async function oe(C,S){return h(),C=fe(C),Qe(C,"customer.query"),le(D({query:C,type:"query",...S}),{logErrors:i})}function J(C){e.set(ge,{...e.get(ge),...C});}async function ne(){let C=await F();if(C)return {...e.get(ge),customerAccessToken:C}}return {login:async C=>{h();let S=new URL(f("AUTH")),x=jr(),w=Je();S.searchParams.set("client_id",t),S.searchParams.set("scope","openid email"),S.searchParams.append("response_type","code"),S.searchParams.append("redirect_uri",P),S.searchParams.set("scope",f("LOGIN_SCOPE")),S.searchParams.append("state",x),S.searchParams.append("nonce",w);let V=ia({contextLanguage:m??null,uiLocalesOverride:C?.uiLocales??null});V!=null&&S.searchParams.append("ui_locales",V);let H=Gr(),k=await Qr(H);return e.set(_,{...e.get(_),codeVerifier:H,state:x,nonce:w,redirectPath:Ye(n.url)||Q(n,"Referer")||l}),S.searchParams.append("code_challenge",k),S.searchParams.append("code_challenge_method","S256"),Re(S.toString())},logout:async C=>{h();let S=e.get(_)?.idToken,x=Mt({requestUrl:g,defaultUrl:g,redirectUrl:C?.postLogoutRedirectUri}),w=S?new URL(`${f("LOGOUT")}?${new URLSearchParams([["id_token_hint",S],["post_logout_redirect_uri",x]]).toString()}`).toString():x;return he(e),Re(w,{headers:C?.headers||{}})},isLoggedIn:U,handleAuthStatus:O,getAccessToken:F,getApiUrl:()=>A,mutate:xe,query:oe,authorize:async()=>{h();let C=d.searchParams.get("code"),S=d.searchParams.get("state");if(!C||!S)throw he(e),new $("Unauthorized","No code or state parameter found in the redirect URL.");if(e.get(_)?.state!==S)throw he(e),new $("Unauthorized","The session state does not match the state parameter. Make sure that the session is configured correctly and passed to `createCustomerAccountClient`.");let x=t,w=new URLSearchParams;w.append("grant_type","authorization_code"),w.append("client_id",x),w.append("redirect_uri",P),w.append("code",C);let V=e.get(_)?.codeVerifier;if(!V)throw new $("Unauthorized","No code verifier found in the session. Make sure that the session is configured correctly and passed to `createCustomerAccountClient`.");w.append("code_verifier",V);let H={"content-type":"application/x-www-form-urlencoded","User-Agent":me,Origin:g},k=q?.();new Date().getTime();let X=f("TOKEN_EXCHANGE"),v=await fetch(X,{method:"POST",headers:H,body:w});if(!v.ok)throw new Response(await v.text(),{status:v.status,headers:{"Content-Type":"text/html; charset=utf-8"}});let{access_token:N,expires_in:De,id_token:_e,refresh_token:et}=await v.json(),Kt=e.get(_)?.nonce,zt=await Kr(_e);if(Kt!==zt)throw new $("Unauthorized",`Returned nonce does not match: ${Kt} !== ${zt}`);let Jt=N;r||(Jt=await Nt(N,t,f("TOKEN_EXCHANGE"),g,{waitUntil:a,stackInfo:k,...ee(n)}));let So=e.get(_)?.redirectPath;return e.set(_,{accessToken:Jt,expiresAt:new Date(new Date().getTime()+(De-120)*1e3).getTime()+"",refreshToken:et,idToken:_e}),Re(So||l)},setBuyer:J,getBuyer:ne,UNSTABLE_setBuyer:C=>{G("[h2:warn:customerAccount] `customerAccount.UNSTABLE_setBuyer` is deprecated. Please use `customerAccount.setBuyer`."),J(C);},UNSTABLE_getBuyer:()=>(G("[h2:warn:customerAccount] `customerAccount.UNSTABLE_getBuyer` is deprecated. Please use `customerAccount.getBuyer`."),ne())}}function sa(e,t){return function(){try{if(!t)throw Error();new URL(e("CA_BASE_URL")),new URL(e("CA_BASE_AUTH_URL"));}catch{console.error(new Error("[h2:error:customerAccount] You do not have the valid credential to use Customer Account API.\nRun `h2 env pull` to link your store credentials."));let o="Internal Server Error";throw new Response(o,{status:500})}}}function ia(e){let t=Xr(e.contextLanguage??null);return Xr(e.uiLocalesOverride)??t??null}function Xr(e){if(e==null)return null;let o=ua(e).toLowerCase().replaceAll("_","-").split("-"),n=o.at(0)??null,a=o.at(1)??null;return a?`${n}-${a.toUpperCase()}`:n}var ca={PT:"PT_PT",ZH:"ZH_CN"};function ua(e){return ca[e]??e}function pa(e){let{env:t,request:r,cache:o,waitUntil:n,i18n:a,session:s,logErrors:c,storefront:i={},customerAccount:u,cart:p={}}=e;s||console.warn("[h2:warn:createHydrogenContext] A session object is required to create hydrogen context."),u?.unstableB2b&&G("[h2:warn:createHydrogenContext] `customerAccount.unstableB2b` is now stable. Please remove the `unstableB2b` option.");let{storefront:l}=$r({cache:o,waitUntil:n,i18n:a,logErrors:c,storefrontHeaders:i.headers||da(r),storefrontApiVersion:i.apiVersion,storefrontId:t.PUBLIC_STOREFRONT_ID,storeDomain:t.PUBLIC_STORE_DOMAIN,privateStorefrontToken:t.PRIVATE_STOREFRONT_API_TOKEN,publicStorefrontToken:t.PUBLIC_STOREFRONT_API_TOKEN}),m=Vt({session:s,request:r,waitUntil:n,logErrors:c,customerApiVersion:u?.apiVersion,authUrl:u?.authUrl,customAuthStatusHandler:u?.customAuthStatusHandler,language:a?.language,customerAccountId:t.PUBLIC_CUSTOMER_ACCOUNT_API_CLIENT_ID,shopId:t.SHOP_ID}),y=kt({getCartId:p.getId||mt(r.headers),setCartId:p.setId||gt(),cartQueryFragment:p.queryFragment,cartMutateFragment:p.mutateFragment,customMethods:p.customMethods,storefront:l,customerAccount:m});return {storefront:l,customerAccount:m,cart:y,env:t,waitUntil:n,session:s}}function da(e){return {requestGroupId:Q(e,"request-id"),buyerIp:Q(e,"oxygen-buyer-ip"),cookie:Q(e,"cookie"),purpose:Q(e,"purpose")}}var Zr=react.createContext(void 0),ga=Zr.Provider,$t=()=>react.useContext(Zr);function ha(e){let t=Je(),r=Ca(t,e);return {nonce:t,header:r,NonceProvider:({children:n})=>react.createElement(ga,{value:t},n)}}function Ca(e,t){let{shop:r,...o}=t??{},n=`'nonce-${e}'`,a=["'self'","'unsafe-inline'","https://cdn.shopify.com"],s=["'self'","https://monorail-edge.shopifysvc.com"];r&&r.checkoutDomain&&s.push(`https://${r.checkoutDomain}`),r&&r.storeDomain&&s.push(`https://${r.storeDomain}`);let i={baseUri:["'self'"],defaultSrc:["'self'",n,"https://cdn.shopify.com","https://shopify.com"],frameAncestors:["'none'"],styleSrc:a,connectSrc:s},u=Object.assign({},i,o);for(let p in i){let l=o[p];p&&l&&(u[p]=Pa(l,i[p]));}return u.scriptSrc instanceof Array?u.scriptSrc=[...u.scriptSrc.filter(p=>!p.startsWith("'nonce")),n]:u.defaultSrc instanceof Array&&(u.defaultSrc=[...u.defaultSrc.filter(p=>!p.startsWith("'nonce")),n]),ma__default.default({directives:u})}function Pa(e,t){let r=typeof t=="string"?[t]:t,o=Array.isArray(e)?e:[String(e)];return Array.isArray(r)?r.every(a=>a==="'none'")?o:[...o,...r]:r}var Ta=react.forwardRef((e,t)=>{let{waitForHydration:r,src:o,...n}=e,a=$t();return r?jsxRuntime.jsx(va,{src:o,options:n}):jsxRuntime.jsx("script",{suppressHydrationWarning:!0,...n,src:o,nonce:a,ref:t})});function va({src:e,options:t}){if(!e)throw new Error("`waitForHydration` with the Script component requires a `src` prop");return hydrogenReact.useLoadScript(e,{attributes:t}),null}function Ea(e){let t=react$1.useFetchers(),r={};for(let{formData:o}of t)if(o?.get("optimistic-identifier")===e)try{if(o.has("optimistic-data")){let n=JSON.parse(String(o.get("optimistic-data")));Object.assign(r,n);}}catch{}return r}function ba({id:e,data:t}){return jsxRuntime.jsxs(jsxRuntime.Fragment,{children:[jsxRuntime.jsx("input",{type:"hidden",name:"optimistic-identifier",value:e}),jsxRuntime.jsx("input",{type:"hidden",name:"optimistic-data",value:JSON.stringify(t)})]})}function Da({connection:e,children:t=()=>(console.warn("<Pagination> requires children to work properly"),null),namespace:r=""}){let [o,n]=react.useState(!1),a=react$1.useNavigation(),s=react$1.useLocation();react$1.useNavigate();react.useEffect(()=>{a.state==="idle"&&n(!1);},[a.state]);let{endCursor:i,hasNextPage:u,hasPreviousPage:p,nextPageUrl:l,nodes:m,previousPageUrl:y,startCursor:d}=_a(e,r),g=react.useMemo(()=>({...s.state,pagination:{...s.state?.pagination||{},[r]:{pageInfo:{endCursor:i,hasPreviousPage:p,hasNextPage:u,startCursor:d},nodes:m}}}),[i,u,p,d,m,r,s.state]),P=react.useMemo(()=>react.forwardRef(function(A,R){return u?react.createElement(react$1.Link,{preventScrollReset:!0,...A,to:l,state:g,replace:!0,ref:R,onClick:()=>n(!0)}):null}),[u,l,g]),f=react.useMemo(()=>react.forwardRef(function(A,R){return p?react.createElement(react$1.Link,{preventScrollReset:!0,...A,to:y,state:g,replace:!0,ref:R,onClick:()=>n(!0)}):null}),[p,y,g]);return t({state:g,hasNextPage:u,hasPreviousPage:p,isLoading:o,nextPageUrl:l,nodes:m,previousPageUrl:y,NextLink:P,PreviousLink:f})}function Xe(e,t){let r=new URLSearchParams(e);return Object.keys(t?.pagination||{}).forEach(n=>{let a=n===""?"":`${n}_`,s=`${a}cursor`,c=`${a}direction`;r.delete(s),r.delete(c);}),r.toString()}function we(e){throw new Error(`The Pagination component requires ${"`"+e+"`"} to be a part of your query. See the guide on how to setup your query to include ${"`"+e+"`"}: https://shopify.dev/docs/custom-storefronts/hydrogen/data-fetching/pagination#setup-the-paginated-query`)}function _a(e,t=""){e.pageInfo||we("pageInfo"),typeof e.pageInfo.startCursor>"u"&&we("pageInfo.startCursor"),typeof e.pageInfo.endCursor>"u"&&we("pageInfo.endCursor"),typeof e.pageInfo.hasNextPage>"u"&&we("pageInfo.hasNextPage"),typeof e.pageInfo.hasPreviousPage>"u"&&we("pageInfo.hasPreviousPage");let r=react$1.useNavigation(),o=react$1.useNavigate(),{state:n,search:a,pathname:s}=react$1.useLocation(),c=t?`${t}_cursor`:"cursor",i=t?`${t}_direction`:"direction",l=new URLSearchParams(a).get(i)==="previous",m=react.useMemo(()=>!globalThis?.window?.__hydrogenHydrated||!n?.pagination?.[t]?.nodes?hydrogenReact.flattenConnection(e):l?[...hydrogenReact.flattenConnection(e),...n.pagination[t].nodes||[]]:[...n.pagination[t].nodes||[],...hydrogenReact.flattenConnection(e)],[n,e,t]),y=react.useMemo(()=>{let f=globalThis?.window?.__hydrogenHydrated,h=n?.pagination?.[t]?.pageInfo,A=!f||h?.startCursor===void 0?e.pageInfo.startCursor:h.startCursor,R=!f||h?.endCursor===void 0?e.pageInfo.endCursor:h.endCursor,D=!f||h?.hasPreviousPage===void 0?e.pageInfo.hasPreviousPage:h.hasPreviousPage,U=!f||h?.hasNextPage===void 0?e.pageInfo.hasNextPage:h.hasNextPage;return n?.pagination?.[t]?.nodes&&(l?(A=e.pageInfo.startCursor,D=e.pageInfo.hasPreviousPage):(R=e.pageInfo.endCursor,U=e.pageInfo.hasNextPage)),{startCursor:A,endCursor:R,hasPreviousPage:D,hasNextPage:U}},[l,n,t,e.pageInfo.hasNextPage,e.pageInfo.hasPreviousPage,e.pageInfo.startCursor,e.pageInfo.endCursor]),d=react.useRef({params:Xe(a,n),pathname:s});react.useEffect(()=>{window.__hydrogenHydrated=!0;},[]),react.useEffect(()=>{let f=Xe(a,n),h=d.current.params;(s!==d.current.pathname||f!==h)&&!(r.state==="idle"&&!r.location)&&(d.current={pathname:s,params:Xe(a,n)},o(`${s}?${Xe(a,n)}`,{replace:!0,preventScrollReset:!0,state:{nodes:void 0,pageInfo:void 0}}));},[s,a,n]);let g=react.useMemo(()=>{let f=new URLSearchParams(a);return f.set(i,"previous"),y.startCursor&&f.set(c,y.startCursor),`?${f.toString()}`},[a,y.startCursor]),P=react.useMemo(()=>{let f=new URLSearchParams(a);return f.set(i,"next"),y.endCursor&&f.set(c,y.endCursor),`?${f.toString()}`},[a,y.endCursor]);return {...y,previousPageUrl:g,nextPageUrl:P,nodes:m}}function La(e,t={pageBy:20}){if(typeof e?.url>"u")throw new Error("getPaginationVariables must be called with the Request object passed to your loader function");let{pageBy:r,namespace:o=""}=t,n=new URLSearchParams(new URL(e.url).search),a=o?`${o}_cursor`:"cursor",s=o?`${o}_direction`:"direction",c=n.get(a)??void 0;return (n.get(s)==="previous"?"previous":"next")==="previous"?{last:r,startCursor:c??null}:{first:r,endCursor:c??null}}function Ma(e,t){let r=react$1.useNavigation(),[o,n]=react.useState([]);if(react.useEffect(()=>{Promise.resolve(t).then(a=>{a&&n(a instanceof Array?a:a.product?.variants?.nodes||[]);}).catch(a=>{reportError(new Error("[h2:error:useOptimisticVariant] An error occurred while resolving the variants for the optimistic product hook.",{cause:a}));});},[JSON.stringify(t)]),r.state==="loading"){let a=new URLSearchParams(r.location.search),s=!1,c=o.find(i=>i.selectedOptions?i.selectedOptions.every(u=>a.get(u.name)===u.value):(s||(s=!0,reportError(new Error("[h2:error:useOptimisticVariant] The optimistic product hook requires your product query to include variants with the selectedOptions field."))),!1));if(c)return {...c,isOptimistic:!0}}return e}function Ba({handle:e,options:t=[],variants:r=[],productPath:o="products",waitForNavigation:n=!1,selectedVariant:a,children:s}){let c=t;c[0]?.values&&(G("[h2:warn:VariantSelector] product.options.values is deprecated. Use product.options.optionValues instead."),c[0]&&!c[0].optionValues&&(c=t.map(d=>({...d,optionValues:d.values?.map(g=>({name:g}))||[]}))));let i=r instanceof Array?r:hydrogenReact.flattenConnection(r),{searchParams:u,path:p,alreadyOnProductPage:l}=Qa(e,o,n),m=c.filter(d=>d?.optionValues?.length===1),y=a?a?.selectedOptions?.reduce((d,g)=>(d[g.name]=g.value,d),{}):{};return react.createElement(react.Fragment,null,...react.useMemo(()=>c.map(d=>{let g,P=[];for(let f of d.optionValues){let h=new URLSearchParams(l?u:void 0);h.set(d.name,f.name),m.forEach(O=>{O.optionValues[0].name&&h.set(O.name,O.optionValues[0].name);});let A=i.find(O=>O?.selectedOptions?.every(F=>(h.get(F?.name)||y?.[F?.name])===F?.value)),R=u.get(d.name);!R&&a&&(R=y?.[d.name]||null);let D=R?R===f.name:!1;D&&(g=f.name);let U="?"+h.toString();P.push({value:f.name,optionValue:f,isAvailable:A?A.availableForSale:!0,to:p+U,search:U,isActive:D,variant:A});}return s({option:{name:d.name,value:g,values:P}})}),[c,i,s]))}var Ga=e=>{if(typeof e?.url>"u")throw new TypeError(`Expected a Request instance, got ${typeof e}`);let t=new URL(e.url).searchParams,r=[];return t.forEach((o,n)=>{r.push({name:n,value:o});}),r};function Qa(e,t,r){let{pathname:o,search:n}=react$1.useLocation(),a=react$1.useNavigation();return react.useMemo(()=>{let s=/(\/[a-zA-Z]{2}-[a-zA-Z]{2}\/)/g.exec(o),c=s&&s.length>0;t=t.startsWith("/")?t.substring(1):t;let i=c?`${s[0]}${t}/${e}`:`/${t}/${e}`;return {searchParams:new URLSearchParams(r||a.state!=="loading"?n:a.location.search),alreadyOnProductPage:i===o,path:i}},[o,n,r,e,t,a])}var Ka=function(e){return jsxRuntime.jsx(hydrogenReact.RichText,{...e,components:{link:({node:t})=>jsxRuntime.jsx(react$1.Link,{to:t.url,title:t.title,target:t.target,prefetch:"intent",children:t.children}),...e.components}})};var za=async function({request:t,context:r}){let o=r.storefront,n=r.customerAccount,a=new URL(t.url);if(!o)throw new Error("GraphiQL: Hydrogen's storefront client must be injected in the loader context.");let s={};if(o){let u="X-Shopify-Storefront-Access-Token";s.storefront={name:"Storefront API",authHeader:u,accessToken:o.getPublicTokenHeaders()[u],apiUrl:o.getApiUrl(),icon:"SF"};}if(n){let u=await(await fetch(a.origin+"/graphiql/customer-account.schema.json")).json(),p=await n.getAccessToken();u&&(s["customer-account"]={name:"Customer Account API",value:u,authHeader:"Authorization",accessToken:p,apiUrl:n.getApiUrl(),icon:"CA"});}let c="https://avatars.githubusercontent.com/u/12972006?s=48&v=4",i=String.raw;return new Response(i`
      <!DOCTYPE html>
      <html lang="en">
        <head>
          <title>GraphiQL</title>
          <link rel="icon" type="image/x-icon" href="${c}" />
          <style>
            body {
              height: 100%;
              margin: 0;
              width: 100%;
              overflow: hidden;
              background-color: hsl(219, 29%, 18%);
            }

            #graphiql {
              height: 100vh;
            }

            #graphiql > .placeholder {
              color: slategray;
              width: fit-content;
              margin: 40px auto;
              font-family: Arial;
            }

            .graphiql-api-toolbar-label {
              position: absolute;
              bottom: -6px;
              right: -4px;
              font-size: 8px;
            }
          </style>

          <script
            crossorigin
            src="https://unpkg.com/react@18/umd/react.development.js"
          ></script>
          <script
            crossorigin
            src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"
          ></script>
          <link
            rel="stylesheet"
            href="https://unpkg.com/graphiql@3/graphiql.min.css"
          />
          <link
            rel="stylesheet"
            href="https://unpkg.com/@graphiql/plugin-explorer/dist/style.css"
          />
        </head>

        <body>
          <div id="graphiql">
            <div class="placeholder">Loading GraphiQL...</div>
          </div>

          <script
            src="https://unpkg.com/graphiql@3/graphiql.min.js"
            type="application/javascript"
            crossorigin="anonymous"
          ></script>
          <script
            src="https://unpkg.com/@graphiql/plugin-explorer/dist/index.umd.js"
            type="application/javascript"
            crossorigin="anonymous"
          ></script>

          <script>
            const windowUrl = new URL(document.URL);
            const startingSchemaKey =
              windowUrl.searchParams.get('schema') || 'storefront';

            let query = '{ shop { name } }';
            if (windowUrl.searchParams.has('query')) {
              query = decodeURIComponent(
                windowUrl.searchParams.get('query') ?? query,
              );
            }

            // Prettify query
            query = GraphiQL.GraphQL.print(GraphiQL.GraphQL.parse(query));

            let variables;
            if (windowUrl.searchParams.has('variables')) {
              variables = decodeURIComponent(
                windowUrl.searchParams.get('variables') ?? '',
              );
            }

            // Prettify variables
            if (variables) {
              variables = JSON.stringify(JSON.parse(variables), null, 2);
            }

            const schemas = ${JSON.stringify(s)};
            let lastActiveTabIndex = -1;
            let lastTabAmount = -1;

            const root = ReactDOM.createRoot(
              document.getElementById('graphiql'),
            );

            root.render(React.createElement(RootWrapper));

            const TAB_STATE_KEY = 'graphiql:tabState';
            const storage = {
              getTabState: () =>
                JSON.parse(localStorage.getItem(TAB_STATE_KEY)),
              setTabState: (state) =>
                localStorage.setItem(TAB_STATE_KEY, JSON.stringify(state)),
            };

            let nextSchemaKey;

            function RootWrapper() {
              const [activeSchema, setActiveSchema] =
                React.useState(startingSchemaKey);

              const schema = schemas[activeSchema];
              if (!schema) {
                throw new Error('No schema found for ' + activeSchema);
              }

              const keys = Object.keys(schemas);

              return React.createElement(
                GraphiQL,
                {
                  fetcher: GraphiQL.createFetcher({
                    url: schema.apiUrl,
                    headers: {[schema.authHeader]: schema.accessToken},
                  }),
                  defaultEditorToolsVisibility: true,
                  query,
                  variables,
                  schema: schema.value,
                  plugins: [GraphiQLPluginExplorer.explorerPlugin()],
                  onTabChange: (state) => {
                    const {activeTabIndex, tabs} = state;
                    const activeTab = tabs[activeTabIndex];

                    if (
                      activeTabIndex === lastActiveTabIndex &&
                      lastTabAmount === tabs.length
                    ) {
                      if (
                        nextSchemaKey &&
                        activeTab &&
                        activeTab.schemaKey !== nextSchemaKey
                      ) {
                        activeTab.schemaKey = nextSchemaKey;
                        nextSchemaKey = undefined;

                        // Sync state to localStorage. GraphiQL resets the state
                        // asynchronously, so we need to do it in a timeout.
                        storage.setTabState(state);
                        setTimeout(() => storage.setTabState(state), 500);
                      }

                      // React rerrendering, skip
                      return;
                    }

                    if (activeTab) {
                      if (!activeTab.schemaKey) {
                        // Creating a new tab
                        if (lastTabAmount < tabs.length) {
                          activeTab.schemaKey = activeSchema;
                          storage.setTabState(state);
                        }
                      }

                      const nextSchema = activeTab.schemaKey || 'storefront';

                      if (nextSchema !== activeSchema) {
                        setActiveSchema(nextSchema);
                      }
                    }

                    lastActiveTabIndex = activeTabIndex;
                    lastTabAmount = tabs.length;
                  },
                  toolbar: {
                    additionalComponent: function () {
                      const schema = schemas[activeSchema];

                      return React.createElement(
                        GraphiQL.React.ToolbarButton,
                        {
                          onClick: () => {
                            const activeKeyIndex = keys.indexOf(activeSchema);
                            nextSchemaKey =
                              keys[(activeKeyIndex + 1) % keys.length];

                            // This triggers onTabChange
                            if (nextSchemaKey) setActiveSchema(nextSchemaKey);
                          },
                          label: 'Toggle between different API schemas',
                        },
                        React.createElement(
                          'div',
                          {
                            key: 'api-wrapper',
                            className: 'graphiql-toolbar-icon',
                            style: {position: 'relative', fontWeight: 'bolder'},
                          },
                          [
                            React.createElement(
                              'div',
                              {key: 'icon', style: {textAlign: 'center'}},
                              [
                                schema.icon,
                                React.createElement(
                                  'div',
                                  {
                                    key: 'icon-label',
                                    className: 'graphiql-api-toolbar-label',
                                  },
                                  'API',
                                ),
                              ],
                            ),
                          ],
                        ),
                      );
                    },
                  },
                },
                [
                  React.createElement(
                    GraphiQL.Logo,
                    {
                      key: 'Logo replacement',
                    },
                    [
                      React.createElement(
                        'div',
                        {
                          key: 'Logo wrapper',
                          style: {display: 'flex', alignItems: 'center'},
                        },
                        [
                          React.createElement(
                            'div',
                            {
                              key: 'api',
                              className: 'graphiql-logo',
                              style: {
                                paddingRight: 0,
                                whiteSpace: 'nowrap',
                              },
                            },
                            [schema.name],
                          ),
                          React.createElement(GraphiQL.Logo, {key: 'logo'}),
                        ],
                      ),
                    ],
                  ),
                ],
              );
            }
          </script>
        </body>
      </html>
    `,{status:200,headers:{"content-type":"text/html"}})};async function Ja(e){let{storefront:t,request:r,noAdminRedirect:o,matchQueryParams:n,response:a=new Response("Not Found",{status:404})}=e,s=new URL(r.url),{pathname:c,searchParams:i}=s,u=i.has("_data");i.delete("redirect"),i.delete("return_to"),i.delete("_data");let p=(n?s.toString().replace(s.origin,""):c).toLowerCase();if(s.pathname==="/admin"&&!o)return Bt(`${t.getShopifyDomain()}/admin`,u,i,n);try{let{urlRedirects:l}=await t.query(Ya,{variables:{query:"path:"+p.replace(/\/+$/,"")}}),m=l?.edges?.[0]?.node?.target;if(m)return Bt(m,u,i,n);let y=Ye(r.url);if(y)return Bt(y,u,i,n)}catch(l){console.error(`Failed to fetch redirects from Storefront API for route ${p}`,l);}return a}var Ht="https://example.com";function Bt(e,t,r,o){let n=new URL(e,Ht);if(!o)for(let[a,s]of r)n.searchParams.append(a,s);return t?new Response(null,{status:200,headers:{"X-Remix-Redirect":n.toString().replace(Ht,""),"X-Remix-Status":"301"}}):new Response(null,{status:301,headers:{location:n.toString().replace(Ht,"")}})}var Ya=`#graphql
  query redirects($query: String) {
    urlRedirects(first: 1, query: $query) {
      edges {
        node {
          target
        }
      }
    }
  }
`;var Xa={"&":"\\u0026",">":"\\u003e","<":"\\u003c","\u2028":"\\u2028","\u2029":"\\u2029"},Za=/[&><\u2028\u2029]/g;function po(e){return e.replace(Za,t=>Xa[t])}var te="Error in SEO input: ",K={title:{validate:e=>{if(typeof e!="string")throw new Error(te.concat("`title` should be a string"));if(typeof e=="string"&&e.length>120)throw new Error(te.concat("`title` should not be longer than 120 characters"));return e}},description:{validate:e=>{if(typeof e!="string")throw new Error(te.concat("`description` should be a string"));if(typeof e=="string"&&e.length>155)throw new Error(te.concat("`description` should not be longer than 155 characters"));return e}},url:{validate:e=>{if(typeof e!="string")throw new Error(te.concat("`url` should be a string"));if(typeof e=="string"&&!e.startsWith("http"))throw new Error(te.concat("`url` should be a valid URL"));return e}},handle:{validate:e=>{if(typeof e!="string")throw new Error(te.concat("`handle` should be a string"));if(typeof e=="string"&&!e.startsWith("@"))throw new Error(te.concat("`handle` should start with `@`"));return e}}};function lo(e){let t=[];for(let r of Object.keys(e))switch(r){case"title":{let o=z(K.title,e.title),n=Qt(e?.titleTemplate,o);if(!n)break;t.push(L("title",{title:n}),L("meta",{property:"og:title",content:n}),L("meta",{name:"twitter:title",content:n}));break}case"description":{let o=z(K.description,e.description);if(!o)break;t.push(L("meta",{name:"description",content:o}),L("meta",{property:"og:description",content:o}),L("meta",{name:"twitter:description",content:o}));break}case"url":{let o=z(K.url,e.url);if(!o)break;let a=o.split("?")[0].replace(/\/$/,"");t.push(L("link",{rel:"canonical",href:a}),L("meta",{property:"og:url",content:a}));break}case"handle":{let o=z(K.handle,e.handle);if(!o)break;t.push(L("meta",{name:"twitter:site",content:o}),L("meta",{name:"twitter:creator",content:o}));break}case"media":{let o,n=re(e.media);for(let a of n)if(typeof a=="string"&&t.push(L("meta",{name:"og:image",content:a})),a&&typeof a=="object"){let s=a.type||"image",c=a?{url:a?.url,secure_url:a?.url,type:Wt(a.url),width:a?.width,height:a?.height,alt:a?.altText}:{};for(let i of Object.keys(c))c[i]&&(o=c[i],t.push(L("meta",{property:`og:${s}:${i}`,content:o},c.url)));}break}case"jsonLd":{let o=re(e.jsonLd),n=0;for(let a of o){if(typeof a!="object")continue;let s=L("script",{type:"application/ld+json",children:JSON.stringify(a,(c,i)=>typeof i=="string"?po(i):i)},`json-ld-${a?.["@type"]||a?.name||n++}`);t.push(s);}break}case"alternates":{let o=re(e.alternates);for(let n of o){if(!n)continue;let{language:a,url:s,default:c}=n,i=a?`${a}${c?"-default":""}`:void 0;t.push(L("link",{rel:"alternate",hrefLang:i,href:s}));}break}case"robots":{if(!e.robots)break;let{maxImagePreview:o,maxSnippet:n,maxVideoPreview:a,noArchive:s,noFollow:c,noImageIndex:i,noIndex:u,noSnippet:p,noTranslate:l,unavailableAfter:m}=e.robots,y=[s&&"noarchive",i&&"noimageindex",p&&"nosnippet",l&&"notranslate",o&&`max-image-preview:${o}`,n&&`max-snippet:${n}`,a&&`max-video-preview:${a}`,m&&`unavailable_after:${m}`],d=(u?"noindex":"index")+","+(c?"nofollow":"follow");for(let g of y)g&&(d+=`,${g}`);t.push(L("meta",{name:"robots",content:d}));break}}return t.flat().sort((r,o)=>r.key.localeCompare(o.key))}function L(e,t,r){let o={tag:e,props:{},key:""};return e==="title"?(o.children=t.title,o.key=Gt(o),o):e==="script"?(o.children=typeof t.children=="string"?t.children:"",o.key=Gt(o,r),delete t.children,o.props=t,o):(o.props=t,Object.keys(o.props).forEach(n=>!o.props[n]&&delete o.props[n]),o.key=Gt(o,r),o)}function Gt(e,t){let{tag:r,props:o}=e;if(r==="title")return "0-title";if(r==="meta"){let n=o.content===t&&typeof o.property=="string"&&!o.property.endsWith("secure_url")&&"0";return [r,...[t,n],o.property||o.name].filter(s=>s).join("-")}return r==="link"?[r,o.rel,o.hrefLang||o.media].filter(a=>a).join("-").replace(/\s+/g,"-"):r==="script"?`${r}-${t}`:`${r}-${o.type}`}function Qt(e,t){if(t)return e?typeof e=="function"?e(t):e.replace("%s",t??""):t}function Wt(e){switch(e&&e.split(".").pop()){case"svg":return "image/svg+xml";case"png":return "image/png";case"gif":return "image/gif";case"swf":return "application/x-shockwave-flash";case"mp3":return "audio/mpeg";case"jpg":case"jpeg":default:return "image/jpeg"}}function re(e){return Array.isArray(e)?e:[e]}function z(e,t){try{return e.validate(t)}catch(r){return console.warn(r.message),t}}function es(...e){let t=[],r=e.reduce((o,n)=>{if(!n)return o;Object.keys(n).forEach(s=>!n[s]&&delete n[s]);let{jsonLd:a}=n;return a?o?.jsonLd?{...o,...n,jsonLd:re(o.jsonLd).concat(a)}:{...o,...n,jsonLd:[a]}:{...o,...n}},{})||{};for(let o of Object.keys(r))switch(o){case"title":{let n=z(K.title,r.title),a=Qt(r?.titleTemplate,n);if(!a)break;t.push({title:a},{property:"og:title",content:a},{property:"twitter:title",content:a});break}case"description":{let n=z(K.description,r.description);if(!n)break;t.push({name:"description",content:n},{property:"og:description",content:n},{property:"twitter:description",content:n});break}case"url":{let n=z(K.url,r.url);if(!n)break;let s=n.split("?")[0].replace(/\/$/,"");t.push({tagName:"link",rel:"canonical",href:s},{property:"og:url",content:s});break}case"handle":{let n=z(K.handle,r.handle);if(!n)break;t.push({property:"twitter:site",content:n},{property:"twitter:creator",content:n});break}case"media":{let n,a=re(r.media);for(let s of a)if(typeof s=="string"&&t.push({property:"og:image",content:s}),s&&typeof s=="object"){let c=s.type||"image",i=s?{url:s?.url,secure_url:s?.url,type:Wt(s.url),width:s?.width,height:s?.height,alt:s?.altText}:{};for(let u of Object.keys(i))i[u]&&(n=i[u],t.push({property:`og:${c}:${u}`,content:n}));}break}case"jsonLd":{let n=re(r.jsonLd);for(let s of n)typeof s!="object"||Object.keys(s).length===0||t.push({"script:ld+json":s});break}case"alternates":{let n=re(r.alternates);for(let a of n){if(!a)continue;let{language:s,url:c,default:i}=a,u=s?`${s}${i?"-default":""}`:void 0;t.push({tagName:"link",rel:"alternate",hrefLang:u,href:c});}break}case"robots":{if(!r.robots)break;let{maxImagePreview:n,maxSnippet:a,maxVideoPreview:s,noArchive:c,noFollow:i,noImageIndex:u,noIndex:p,noSnippet:l,noTranslate:m,unavailableAfter:y}=r.robots,d=[c&&"noarchive",u&&"noimageindex",l&&"nosnippet",m&&"notranslate",n&&`max-image-preview:${n}`,a&&`max-snippet:${a}`,s&&`max-video-preview:${s}`,y&&`unavailable_after:${y}`],g=(p?"noindex":"index")+","+(i?"nofollow":"follow");for(let P of d)P&&(g+=`,${P}`);t.push({name:"robots",content:g});break}}return t}var us=react.lazy(()=>Promise.resolve().then(()=>(go(),mo)));function ps({debug:e}){let t=react$1.useMatches(),r=react$1.useLocation();console.warn("[h2:warn:Seo] The `<Seo/>` component is deprecated. Use `getSeoMeta` instead.\nSee: https://shopify.dev/docs/api/hydrogen/2025-01/utilities/getseometa");let o=react.useMemo(()=>t.flatMap(s=>{let{handle:c,...i}=s,u={...i,...r},p=c?.seo,l=i?.data?.seo;return !p&&!l?[]:p?Ze(p,u):[l]}).reduce((s,c)=>{Object.keys(c).forEach(u=>!c[u]&&delete c[u]);let{jsonLd:i}=c;return i?s?.jsonLd?Array.isArray(i)?{...s,...c,jsonLd:[...s.jsonLd,...i]}:{...s,...c,jsonLd:[...s.jsonLd,i]}:{...s,...c,jsonLd:[i]}:{...s,...c}},{}),[t,r]),{html:n,loggerMarkup:a}=react.useMemo(()=>{let s=lo(o),c=s.map(u=>u.tag==="script"?react.createElement(u.tag,{...u.props,key:u.key,dangerouslySetInnerHTML:{__html:u.children}}):react.createElement(u.tag,{...u.props,key:u.key},u.children)),i=react.createElement(react.Suspense,{fallback:null},react.createElement(us,{headTags:s}));return {html:c,loggerMarkup:i}},[o]);return react.createElement(react.Fragment,null,n,e&&a)}function Ze(e,...t){if(e instanceof Function)return Ze(e(...t),...t);let r={};return Array.isArray(e)?(r=e.reduce((o,n)=>[...o,Ze(n)],[]),r):e instanceof Object?(Object.entries(e).forEach(([n,a])=>{r[n]=Ze(a,...t);}),r):e}function ls(e){return jsxRuntime.jsx(hydrogenReact.ShopPayButton,{channel:"hydrogen",...e})}var fs=`<?xml version="1.0" encoding="UTF-8"?>
<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
`,ms=`
</sitemapindex>`,Co=`<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">`,Po="</urlset>";async function gs(e){let{storefront:t,request:r,types:o=["products","pages","collections","metaObjects","articles","blogs"],customChildSitemaps:n=[]}=e;if(!r||!r.url)throw new Error("A request object is required to generate a sitemap index");if(!t||!t.query)throw new Error("A storefront client is required to generate a sitemap index");let a=await t.query(Rs);if(!a)throw console.warn("[h2:sitemap:warning] Sitemap index is available in API version 2024-10 and later"),new Response("Sitemap index not found.",{status:404});let s=new URL(r.url).origin,c=fs+o.map(i=>{if(!a[i])throw new Error(`[h2:sitemap:error] No data found for type ${i}. Check types passed to \`getSitemapIndex\``);return Cs(i,a[i].pagesCount.count,s)}).join(`
`)+n.map(i=>"  <sitemap><loc>"+(s+(i.startsWith("/")?i:"/"+i))+"</loc></sitemap>").join(`
`)+ms;return new Response(c,{headers:{"Content-Type":"application/xml","Cache-Control":`max-age=${60*60*24}`}})}async function hs(e){let{storefront:t,request:r,params:o,getLink:n,locales:a=[],getChangeFreq:s,noItemsFallback:c="/"}=e;if(!o)throw new Error("[h2:sitemap:error] Remix params object is required to generate a sitemap");if(!r||!r.url)throw new Error("A request object is required to generate a sitemap");if(!t||!t.query)throw new Error("A storefront client is required to generate a index");if(!n)throw new Error("A `getLink` function to generate each resource is required to build a sitemap");if(!o.type||!o.page)throw new Response("No data found",{status:404});let i=o.type,u=ws[i];if(!u)throw new Response("Not found",{status:404});let p=await t.query(u,{variables:{page:parseInt(o.page,10)}});if(!p)throw console.warn("[h2:sitemap:warning] Sitemap is available in API version 2024-10 and later"),new Response("Sitemap not found.",{status:404});let l=new URL(r.url).origin,m="";return p?.sitemap?.resources?.items?.length?m=Co+p.sitemap.resources.items.map(y=>Ps({getChangeFreq:s,url:n({type:y.type??i,baseUrl:l,handle:y.handle}),type:i,getLink:n,updatedAt:y.updatedAt,handle:y.handle,metaobjectType:y.type,locales:a,baseUrl:l})).join(`
`)+Po:m=Co+`
  <url><loc>${l+c}</loc></url>
`+Po,new Response(m,{headers:{"Content-Type":"application/xml","Cache-Control":`max-age=${60*60*24}`}})}function Cs(e,t,r){let o="";for(let n=1;n<=t;n++)o+=`  <sitemap><loc>${r}/sitemap/${e}/${n}.xml</loc></sitemap>
`;return o}function Ps({url:e,updatedAt:t,locales:r,type:o,getLink:n,baseUrl:a,handle:s,getChangeFreq:c,metaobjectType:i}){return `<url>
  <loc>${e}</loc>
  <lastmod>${t}</lastmod>
  <changefreq>${c?c({type:i??o,handle:s}):"weekly"}</changefreq>
${r.map(u=>Ss(n({type:i??o,baseUrl:a,handle:s,locale:u}),u)).join(`
`)}
</url>
  `.trim()}function Ss(e,t){return `  <xhtml:link rel="alternate" hreflang="${t}" href="${e}" />`}var As=`#graphql
    query SitemapProducts($page: Int!) {
      sitemap(type: PRODUCT) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,Ts=`#graphql
    query SitemapCollections($page: Int!) {
      sitemap(type: COLLECTION) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,vs=`#graphql
    query SitemapArticles($page: Int!) {
      sitemap(type: ARTICLE) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,Is=`#graphql
    query SitemapPages($page: Int!) {
      sitemap(type: PAGE) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,Es=`#graphql
    query SitemapBlogs($page: Int!) {
      sitemap(type: BLOG) {
        resources(page: $page) {
          items {
            handle
            updatedAt
          }
        }
      }
    }
`,bs=`#graphql
    query SitemapMetaobjects($page: Int!) {
      sitemap(type: METAOBJECT) {
        resources(page: $page) {
          items {
            handle
            updatedAt
            ... on SitemapResourceMetaobject {
              type
            }
          }
        }
      }
    }
`,Rs=`#graphql
query SitemapIndex {
  products: sitemap(type: PRODUCT) {
    pagesCount {
      count
    }
  }
  collections: sitemap(type: COLLECTION) {
    pagesCount {
      count
    }
  }
  articles: sitemap(type: ARTICLE) {
    pagesCount {
      count
    }
  }
  pages: sitemap(type: PAGE) {
    pagesCount {
      count
    }
  }
  blogs: sitemap(type: BLOG) {
    pagesCount {
      count
    }
  }
  metaObjects: sitemap(type: METAOBJECT) {
    pagesCount {
      count
    }
  }
}
`,ws={products:As,articles:vs,collections:Ts,pages:Is,blogs:Es,metaObjects:bs};//! @see https://shopify.dev/docs/api/storefront/latest/queries/cart
//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartCreate
//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartLinesAdd
//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartLinesUpdate
//! @see: https://shopify.dev/docs/api/storefront/latest/mutations/cartLinesRemove
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartDiscountCodesUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartBuyerIdentityUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartNoteUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartSelectedDeliveryOptionsUpdate
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartMetafieldsSet
//! @see https://shopify.dev/docs/api/storefront/2025-01/mutations/cartMetafieldDelete
//! @see https://shopify.dev/docs/api/storefront/latest/mutations/cartGiftCardCodesUpdate

Object.defineProperty(exports, "AnalyticsEventName", {
  enumerable: true,
  get: function () { return hydrogenReact.AnalyticsEventName; }
});
Object.defineProperty(exports, "AnalyticsPageType", {
  enumerable: true,
  get: function () { return hydrogenReact.AnalyticsPageType; }
});
Object.defineProperty(exports, "ExternalVideo", {
  enumerable: true,
  get: function () { return hydrogenReact.ExternalVideo; }
});
Object.defineProperty(exports, "IMAGE_FRAGMENT", {
  enumerable: true,
  get: function () { return hydrogenReact.IMAGE_FRAGMENT; }
});
Object.defineProperty(exports, "Image", {
  enumerable: true,
  get: function () { return hydrogenReact.Image; }
});
Object.defineProperty(exports, "MediaFile", {
  enumerable: true,
  get: function () { return hydrogenReact.MediaFile; }
});
Object.defineProperty(exports, "ModelViewer", {
  enumerable: true,
  get: function () { return hydrogenReact.ModelViewer; }
});
Object.defineProperty(exports, "Money", {
  enumerable: true,
  get: function () { return hydrogenReact.Money; }
});
Object.defineProperty(exports, "ShopifySalesChannel", {
  enumerable: true,
  get: function () { return hydrogenReact.ShopifySalesChannel; }
});
Object.defineProperty(exports, "Video", {
  enumerable: true,
  get: function () { return hydrogenReact.Video; }
});
Object.defineProperty(exports, "customerAccountApiCustomScalars", {
  enumerable: true,
  get: function () { return hydrogenReact.customerAccountApiCustomScalars; }
});
Object.defineProperty(exports, "decodeEncodedVariant", {
  enumerable: true,
  get: function () { return hydrogenReact.decodeEncodedVariant; }
});
Object.defineProperty(exports, "flattenConnection", {
  enumerable: true,
  get: function () { return hydrogenReact.flattenConnection; }
});
Object.defineProperty(exports, "getAdjacentAndFirstAvailableVariants", {
  enumerable: true,
  get: function () { return hydrogenReact.getAdjacentAndFirstAvailableVariants; }
});
Object.defineProperty(exports, "getClientBrowserParameters", {
  enumerable: true,
  get: function () { return hydrogenReact.getClientBrowserParameters; }
});
Object.defineProperty(exports, "getProductOptions", {
  enumerable: true,
  get: function () { return hydrogenReact.getProductOptions; }
});
Object.defineProperty(exports, "getShopifyCookies", {
  enumerable: true,
  get: function () { return hydrogenReact.getShopifyCookies; }
});
Object.defineProperty(exports, "isOptionValueCombinationInEncodedVariant", {
  enumerable: true,
  get: function () { return hydrogenReact.isOptionValueCombinationInEncodedVariant; }
});
Object.defineProperty(exports, "mapSelectedProductOptionToObject", {
  enumerable: true,
  get: function () { return hydrogenReact.mapSelectedProductOptionToObject; }
});
Object.defineProperty(exports, "parseGid", {
  enumerable: true,
  get: function () { return hydrogenReact.parseGid; }
});
Object.defineProperty(exports, "parseMetafield", {
  enumerable: true,
  get: function () { return hydrogenReact.parseMetafield; }
});
Object.defineProperty(exports, "sendShopifyAnalytics", {
  enumerable: true,
  get: function () { return hydrogenReact.sendShopifyAnalytics; }
});
Object.defineProperty(exports, "storefrontApiCustomScalars", {
  enumerable: true,
  get: function () { return hydrogenReact.storefrontApiCustomScalars; }
});
Object.defineProperty(exports, "useLoadScript", {
  enumerable: true,
  get: function () { return hydrogenReact.useLoadScript; }
});
Object.defineProperty(exports, "useMoney", {
  enumerable: true,
  get: function () { return hydrogenReact.useMoney; }
});
Object.defineProperty(exports, "useSelectedOptionInUrlParam", {
  enumerable: true,
  get: function () { return hydrogenReact.useSelectedOptionInUrlParam; }
});
Object.defineProperty(exports, "useShopifyCookies", {
  enumerable: true,
  get: function () { return hydrogenReact.useShopifyCookies; }
});
exports.Analytics = cn;
exports.AnalyticsEvent = M;
exports.CacheCustom = dt;
exports.CacheLong = pt;
exports.CacheNone = ct;
exports.CacheShort = se;
exports.CartForm = j;
exports.InMemoryCache = ft;
exports.OptimisticInput = ba;
exports.Pagination = Da;
exports.RichText = Ka;
exports.Script = Ta;
exports.Seo = ps;
exports.ShopPayButton = ls;
exports.VariantSelector = Ba;
exports.cartAttributesUpdateDefault = Dt;
exports.cartBuyerIdentityUpdateDefault = wt;
exports.cartCreateDefault = vt;
exports.cartDiscountCodesUpdateDefault = Rt;
exports.cartGetDefault = Tt;
exports.cartGetIdDefault = mt;
exports.cartGiftCardCodesUpdateDefault = Ut;
exports.cartLinesAddDefault = It;
exports.cartLinesRemoveDefault = bt;
exports.cartLinesUpdateDefault = Et;
exports.cartMetafieldDeleteDefault = Lt;
exports.cartMetafieldsSetDefault = _t;
exports.cartNoteUpdateDefault = Ot;
exports.cartSelectedDeliveryOptionsUpdateDefault = xt;
exports.cartSetIdDefault = gt;
exports.changelogHandler = Yn;
exports.createCartHandler = kt;
exports.createContentSecurityPolicy = ha;
exports.createCustomerAccountClient = Vt;
exports.createHydrogenContext = pa;
exports.createStorefrontClient = $r;
exports.createWithCache = Pn;
exports.formatAPIResult = T;
exports.generateCacheControlHeader = Te;
exports.getPaginationVariables = La;
exports.getSelectedProductOptions = Ga;
exports.getSeoMeta = es;
exports.getShopAnalytics = an;
exports.getSitemap = hs;
exports.getSitemapIndex = gs;
exports.graphiqlLoader = za;
exports.storefrontRedirect = Ja;
exports.useAnalytics = B;
exports.useCustomerPrivacy = rt;
exports.useNonce = $t;
exports.useOptimisticCart = zn;
exports.useOptimisticData = Ea;
exports.useOptimisticVariant = Ma;
