// @envelop/core/esm/create.js
import { getInstrumented } from '@envelop/instruments';
import { createEnvelopOrchestrator } from './orchestrator.js';
function notEmpty(value) {
    return value != null;
}
export function envelop(options) {
    const plugins = options.plugins.filter(notEmpty);
    const orchestrator = createEnvelopOrchestrator({
        plugins,
    });
    const instruments = orchestrator.instruments;
    const getEnveloped = (context = {}) => {
        const instrumented = getInstrumented({ context });
        const typedOrchestrator = orchestrator;
        instrumented.fn(instruments?.init, orchestrator.init)(context);
        return {
            parse: instrumented.fn(instruments?.parse, typedOrchestrator.parse(context)),
            validate: instrumented.fn(instruments?.validate, typedOrchestrator.validate(context)),
            contextFactory: instrumented.fn(instruments?.context, typedOrchestrator.contextFactory(context)),
            execute: instrumented.asyncFn(instruments?.execute, typedOrchestrator.execute),
            subscribe: instrumented.asyncFn(instruments?.subscribe, typedOrchestrator.subscribe),
            schema: typedOrchestrator.getCurrentSchema(),
        };
    };
    getEnveloped._plugins = plugins;
    return getEnveloped;
}
